<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Voronota version 1.29</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { color: #008000; } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { color: #008000; font-weight: bold; } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <style type="text/css">
  a { color: #0000CC; }
  td { padding-right: 1em; }
  pre { background-color: #DDDDDD; padding: 1em; }
  div#TOC > ul > li > ul > li ul { display: none; }
  </style>
</head>
<body>
<h1>Voronota version 1.29</h1>
<h2>Quick links</h2>
<ul>
<li><a href="./expansion_js/index.html">Voronota-JS</a> (for advanced scripting using JavaScript)</li>
<li><a href="./expansion_lt/index.html">Voronota-LT</a> (for faster and parallelizable computation of tessellation-derived contact areas)</li>
<li><a href="./expansion_gl/index.html">Voronota-GL</a> (for advanced scripting and visualization)</li>
<li><a href="./expansion_gl/web/index.html">Web Voronota-GL</a> (online version of Voronota-GL)</li>
</ul>

<h2>Table of contents</h2>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#about-voronota">About Voronota</a></li>
<li><a href="#quick-install-guide">Quick install guide</a></li>
<li><a href="#about-voronota-expansions">About Voronota expansions</a>
<ul>
<li><a href="#voronota-js">Voronota-JS</a></li>
<li><a href="#voronota-lt">Voronota-LT</a></li>
<li><a href="#voronota-gl">Voronota-GL</a></li>
</ul></li>
<li><a href="#getting-the-latest-version">Getting the latest version</a></li>
<li><a href="#building-from-source-code">Building from source code</a>
<ul>
<li><a href="#requirements">Requirements</a></li>
<li><a href="#using-cmake">Using CMake</a></li>
<li><a href="#using-c-compiler-directly">Using C++ compiler directly</a></li>
<li><a href="#enabling-openmp">Enabling OpenMP</a></li>
<li><a href="#enabling-mpi">Enabling MPI</a></li>
<li><a href="#tr1-usage-switch">TR1 usage switch</a></li>
</ul></li>
<li><a href="#basic-usage-example">Basic usage example</a>
<ul>
<li><a href="#computing-voronoi-vertices">Computing Voronoi vertices</a></li>
<li><a href="#computing-inter-atom-contacts">Computing inter-atom contacts</a></li>
<li><a href="#computing-annotated-inter-atom-contacts">Computing annotated inter-atom contacts</a></li>
<li><a href="#querying-annotated-inter-atom-contacts">Querying annotated inter-atom contacts</a></li>
<li><a href="#getting-help-in-command-line">Getting help in command line</a></li>
</ul></li>
<li><a href="#command-reference">Command reference</a>
<ul>
<li><a href="#list-of-all-commands">List of all commands</a></li>
<li><a href="#command-get-balls-from-atoms-file">Command ‘get-balls-from-atoms-file’</a></li>
<li><a href="#command-calculate-vertices">Command ‘calculate-vertices’</a></li>
<li><a href="#command-calculate-vertices-in-parallel">Command ‘calculate-vertices-in-parallel’</a></li>
<li><a href="#command-calculate-contacts">Command ‘calculate-contacts’</a></li>
<li><a href="#command-query-balls">Command ‘query-balls’</a></li>
<li><a href="#command-query-contacts">Command ‘query-contacts’</a></li>
<li><a href="#command-draw-contacts">Command ‘draw-contacts’</a></li>
<li><a href="#command-score-contacts-energy">Command ‘score-contacts-energy’</a></li>
<li><a href="#command-score-contacts-quality">Command ‘score-contacts-quality’</a></li>
<li><a href="#command-score-contacts-potential">Command ‘score-contacts-potential’</a></li>
<li><a href="#command-compare-contacts">Command ‘compare-contacts’</a></li>
<li><a href="#command-write-balls-to-atoms-file">Command ‘write-balls-to-atoms-file’</a></li>
<li><a href="#command-query-balls-clashes">Command ‘query-balls-clashes’</a></li>
<li><a href="#command-run-script">Command ‘run-script’</a></li>
<li><a href="#command-expand-descriptors">Command ‘expand-descriptors’</a></li>
</ul></li>
<li><a href="#wrapper-scripts">Wrapper scripts</a>
<ul>
<li><a href="#voromqa-method-script">VoroMQA method script</a></li>
<li><a href="#cad-score-method-script">CAD-score method script</a></li>
<li><a href="#contacts-calculation-convenience-script">Contacts calculation convenience script</a></li>
<li><a href="#volumes-calculation-convenience-script">Volumes calculation convenience script</a></li>
<li><a href="#pocket-analysis-script">Pocket analysis script</a></li>
<li><a href="#membrane-fitting-script">Membrane fitting script</a></li>
</ul></li>
</ul>
</nav>
<h1 id="about-voronota">About Voronota</h1>
<p>The analysis of macromolecular structures often requires a comprehensive definition of atomic neighborhoods. Such a definition can be based on the Voronoi diagram of balls, where each ball represents an atom of some van der Waals radius. Voronota is a software tool for finding all the vertices of the Voronoi diagram of balls. Such vertices correspond to the centers of the empty tangent spheres defined by quadruples of balls. Voronota is especially suitable for processing three-dimensional structures of biological macromolecules such as proteins and RNA.</p>
<p>Since version 1.2 Voronota also uses the Voronoi vertices to construct inter-atom contact surfaces and solvent accessible surfaces. Voronota provides tools to query contacts, generate contacts graphics, compare contacts and evaluate quality of protein structural models using contacts.</p>
<p>Voronota is developed by Kliment Olechnovic (kliment.olechnovic@bti.vu.lt).</p>
<h1 id="quick-install-guide">Quick install guide</h1>
<p>Below are several commands that install the latest version of Voronota and its expansions for the command line use in Unix-like systems.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download the latest package</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://github.com/kliment-olechnovic/voronota/releases/download/v1.29.4198/voronota_1.29.4198.tar.gz</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># unpack the package</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xf</span> ./voronota_1.29.4198.tar.gz</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># change to the package directory</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ./voronota_1.29.4198</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># run CMake</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cmake</span> . <span class="at">-DEXPANSION_JS</span><span class="op">=</span>ON <span class="at">-DEXPANSION_LT</span><span class="op">=</span>ON</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># compile everything</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">make</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># install everything</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">sudo</span> make install</span></code></pre></div>
<p>The installed files are listed in the ‘install_manifest.txt’ file. They can be uninstalled by running</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sudo</span> xargs rm <span class="op">&lt;</span> ./install_manifest.txt</span></code></pre></div>
<h1 id="about-voronota-expansions">About Voronota expansions</h1>
<p>Currently there are three expansions of Voronota: Voronota-JS, Voronota-LT and Voronota-GL.</p>
<p>The expansions need to be built separately in their subdirectories. Alternatively, they can be built by appending <code>-DEXPANSION_JS=ON</code> and/or <code>-DEXPANSION_LT=ON</code> and/or <code>-DEXPANSION_GL=ON</code> to the CMake command call.</p>
<p>The expansions have separate dedicated documentation pages.</p>
<h2 id="voronota-js">Voronota-JS</h2>
<p><a href="./expansion_js/index.html">Voronota-JS</a> expansion is located in the ‘expansion_js’ subdirectory of the Voronota package.</p>
<p>Voronota-JS provides a way to write JavaScript scripts for the comprehensive analysis of macromolecular structures, including the Voronoi tesselation-based analysis. Currently, the Voronota-JS package contains several executables:</p>
<ul>
<li>“voronota-js” - core engine that executes JavaScript scripts.</li>
<li>“voronota-js-voromqa” - wrapper to a voronota-js program for computing VoroMQA scores, both old and new (developed for CASP14).</li>
<li>“voronota-js-only-global-voromqa” - wrapper to a voronota-js program for computing only global VoroMQA scores with fast caching.</li>
<li>“voronota-js-membrane-voromqa” - wrapper to a voronota-js program for the VoroMQA-based analysis and assessment of membrane protein structural models.</li>
<li>“voronota-js-ifeatures-voromqa” - wrapper to a voronota-js program for the computation of multiple VoroMQA-based features of protein-protein complexes.</li>
<li>“voronota-js-fast-iface-voromqa” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface VoroMQA energy.</li>
<li>“voronota-js-fast-iface-cadscore” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface CAD-score.</li>
<li>“voronota-js-fast-iface-cadscore-matrix” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface CAD-score matrix.</li>
<li>“voronota-js-fast-iface-data-graph” - wrapper to a voronota-js program for the computation of interface graphs used by the VoroIF-GNN method.</li>
<li>“voronota-js-voroif-gnn” - wrapper to a voronota-js program and GNN inference scripts that run the VoroIF-GNN method for scoring models of protein-protein complexes (developed for CASP15).</li>
<li>“voronota-js-ligand-cadscore” - wrapper to a voronota-js program for the computation of protein-ligand variation of CAD-score (developed to analyze protein-ligand models from CASP15).</li>
</ul>
<h2 id="voronota-lt">Voronota-LT</h2>
<p><a href="./expansion_lt/index.html">Voronota-LT</a> expansion is located in the ‘expansion_lt’ subdirectory of the Voronota package.</p>
<p>Voronota-LT (pronounced ‘voronota lite’) is an alternative version of Voronota for constructing tessellation-derived atomic contact areas and volumes. Voronota-LT was written from scratch and does not use any external code, even from the core Voronota. The primary motivation for creating Voronota-LT was drastically increasing the speed of computing tessellation-based atom-atom contact areas and atom solvent-accessible surface areas. Like Voronota, Voronota-LT can compute contact areas derived from the additively weighted Voronoi tessellation, but the main increase in speed comes when utilizing a simpler, radical tessellation variant, also known as Laguerre-Laguerre tessellation or power diagram.</p>
<h2 id="voronota-gl">Voronota-GL</h2>
<p><a href="./expansion_gl/index.html">Voronota-GL</a> expansion is located in the ‘expansion_gl’ subdirectory of the Voronota package</p>
<p>Voronota-GL is a visual tool for the comprehensive interactive analysis of macromolecular structures, including the Voronoi tesselation-based analysis.</p>
<h1 id="getting-the-latest-version">Getting the latest version</h1>
<p>Download the latest archive from the official downloads page: <a href="https://github.com/kliment-olechnovic/voronota/releases">https://github.com/kliment-olechnovic/voronota/releases</a>.</p>
<p>The archive contains ready-to-use statically compiled ‘voronota’ program for 64 bit Linux systems. This executable can be rebuilt from the provided source code to work on any modern Linux, macOS or Windows operating systems.</p>
<p>On Ubuntu 18.04 and newer it is possible to install Voronota using ‘apt’ command:</p>
<pre><code>sudo apt install voronota</code></pre>
<p>On Windows 10 operating system the easiest way to run Voronota is to use Windows Subsystem for Linux (WSL).</p>
<h1 id="building-from-source-code">Building from source code</h1>
<h2 id="requirements">Requirements</h2>
<p>Voronota has no required external dependencies, only a standard-compliant C++ compiler is needed to build it.</p>
<h2 id="using-cmake">Using CMake</h2>
<p>You can build using CMake for makefile generation. Starting in the directory containing “CMakeLists.txt” file, run the sequence of commands:</p>
<pre><code>cmake ./
make</code></pre>
<p>Alternatively, to keep files more organized, CMake can be run in a separate “build” directory:</p>
<pre><code>mkdir build
cd build
cmake ../
make
cp ./voronota ../voronota</code></pre>
<h2 id="using-c-compiler-directly">Using C++ compiler directly</h2>
<p>For example, “voronota” executable can be built from the sources in “src” directory using GNU C++ compiler:</p>
<pre><code>g++ -O3 -std=c++11 -o voronota $(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<h2 id="enabling-openmp">Enabling OpenMP</h2>
<p>To allow the usage of OpenMP when calling the “calculate-vertices-in-parallel” command, the “-fopenmp” flag needs to be set when building.</p>
<p>When building using C++ compiler directly, just add “-fopenmp”:</p>
<pre><code>g++ -O3 -std=c++11 -fopenmp -o voronota $(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<p>When using CMake, set the CMAKE_CXX_FLAGS variable:</p>
<pre><code>cmake -DCMAKE_CXX_FLAGS=&quot;-fopenmp&quot; ./
make</code></pre>
<h2 id="enabling-mpi">Enabling MPI</h2>
<p>To allow the usage of MPI when calling the “calculate-vertices-in-parallel” command, you can use mpic++ compiler wrapper and define “ENABLE_MPI” macro when buiding:</p>
<pre><code>mpic++ -O3 -std=c++11 -DENABLE_MPI -o voronota ./$(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<h2 id="tr1-usage-switch">TR1 usage switch</h2>
<p>Voronota can be built with either modern C++ compilers or pre-C++11 compilers that support C++ Technical Report 1 (TR1) features. The voronota code has preprocessor-based checks to find out if C++ TR1 namespace is available and needs to be used. If compilation fails, it may mean that these checks failed. To troubleshoot this, try setting the value of the “USE_TR1” macro to 0 (to not use TR1 and to rely on C++11 standard) or 1 (to use TR1) when compiling, for example:</p>
<pre><code>g++ -O3 -DUSE_TR1=1 -o voronota $(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<p>or</p>
<pre><code>g++ -O3 -std=c++11 -DUSE_TR1=0 -o voronota $(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<h1 id="basic-usage-example">Basic usage example</h1>
<h2 id="computing-voronoi-vertices">Computing Voronoi vertices</h2>
<p>Here is a basic example of computing Voronoi vertices for a structure in a PDB file:</p>
<pre><code>./voronota get-balls-from-atoms-file &lt; input.pdb &gt; balls.txt
./voronota calculate-vertices &lt; balls.txt &gt; vertices.txt</code></pre>
<p>The first command reads a PDB file “input.pdb” and outputs a file “balls.txt” that contains balls corresponding to the atoms in “input.pdb” (by default, Voronota ignores all heteroatoms and all hydrogen atoms when reading PDB files: this behavior can be altered using command-line options). The second command reads “balls.txt” and outputs a file “vertices.txt” that contains a quadruples and empty tangent spheres that correspond to the vertices of the Voronoi diagram of the input balls. The formats of “balls.txt” and “vertices.txt” are described below.</p>
<p>In “balls.txt” the line format is “x y z r # comments”. The first four values (x, y, z, r) are atomic ball coordinates and radius. Comments are not needed for further calculations, they are to assist human readers. For example, below is a part of some possible “balls.txt”:</p>
<pre><code>28.888 9.409 52.301 1.7 # 1 A 2 SER N
27.638 10.125 52.516 1.9 # 2 A 2 SER CA
26.499 9.639 51.644 1.75 # 3 A 2 SER C
26.606 8.656 50.915 1.49 # 4 A 2 SER O
27.783 11.635 52.378 1.91 # 5 A 2 SER CB
27.69 12.033 51.012 1.54 # 6 A 2 SER OG</code></pre>
<p>In “vertices.txt” the line format is “q1 q2 q3 q4 x y z r”. The first four numbers (q1, q2, q3, q4) are numbers of atomic records in “balls.txt”, starting from 0. The remaining four values (x, y, z, r) are the coordinates and the radius of an empty tangent sphere of the quadruple of atoms. For example, below is a part of some possible “vertices.txt”:</p>
<pre><code>0 1 2 3 27.761 8.691 51.553 -0.169
0 1 2 23 28.275 9.804 50.131 0.588
0 1 3 1438 24.793 -3.225 60.761 14.047
0 1 4 5 28.785 10.604 50.721 0.283
0 1 4 1453 30.018 10.901 55.386 1.908
0 1 5 23 28.544 10.254 50.194 0.595</code></pre>
<h2 id="computing-inter-atom-contacts">Computing inter-atom contacts</h2>
<p>Taking the “balls.txt” file described in the previous section, here is a basic example of computing inter-atom contacts:</p>
<pre><code>./voronota calculate-contacts &lt; balls.txt &gt; contacts.txt</code></pre>
<p>In “contacts.txt” file the line format is “b1 b2 area”. The first two numbers (b1 and b2) are numbers of atomic records in “balls.txt”, starting from 0. If b1 does not equal b2, then the ‘area’ value is the area of contact between atoms b1 and b2. If b1 equals b2, then the ‘area’ value is the solvent-accessible area of atom b1. For example, below is a part of some possible “contacts.txt”:</p>
<pre><code>0 0 35.440
0 1 15.908
0 2 0.167
0 3 7.025
0 4 7.021
0 5 0.624
0 23 2.849
0 25 0.008
0 26 11.323
0 1454 0.021
1 1 16.448
1 2 11.608
1 3 0.327
1 4 14.170
1 5 0.820
1 6 3.902
1 23 0.081
2 2 3.591
2 3 11.714
2 4 0.305
2 5 2.019</code></pre>
<h2 id="computing-annotated-inter-atom-contacts">Computing annotated inter-atom contacts</h2>
<p>Here is a basic example of computing annotated inter-atom contacts:</p>
<pre><code>./voronota get-balls-from-atoms-file --annotated &lt; input.pdb &gt; annotated_balls.txt
./voronota calculate-contacts --annotated &lt; annotated_balls.txt &gt; annotated_contacts.txt</code></pre>
<p>In “annotated_contacts.txt” the line format is “annotation1 annotation2 area distance tags adjuncts [graphics]”. The strings ‘annotation1’ and ‘annotation2’ describe contacting atoms, the ‘area’ value is the area of contact between the two atoms, the ‘distance’ value is the distance between the centers of the contacting atoms. If ‘annotation2’ contains string “solvent”, then the ‘area’ value is the solvent-accessible area of the atom described by ‘annotation1’. The remaining part of the line is used by Voronota querying and drawing commands that are not covered in this section. Below is a part of some possible “annotated_contacts.txt” (with text fields aligned using “column -t” command):</p>
<pre><code>c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;        15.908  1.456  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;2&gt;a&lt;3&gt;R&lt;SER&gt;A&lt;C&gt;         0.167   2.488  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;2&gt;a&lt;4&gt;R&lt;SER&gt;A&lt;O&gt;         7.025   2.774  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;2&gt;a&lt;5&gt;R&lt;SER&gt;A&lt;CB&gt;        7.021   2.486  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;2&gt;a&lt;6&gt;R&lt;SER&gt;A&lt;OG&gt;        0.624   3.159  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;5&gt;a&lt;24&gt;R&lt;GLU&gt;A&lt;CB&gt;       2.849   4.628  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;5&gt;a&lt;26&gt;R&lt;GLU&gt;A&lt;CD&gt;       0.008   4.792  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;5&gt;a&lt;27&gt;R&lt;GLU&gt;A&lt;OE1&gt;      11.323  3.932  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;A&gt;r&lt;194&gt;a&lt;1501&gt;R&lt;LEU&gt;A&lt;CD2&gt;  0.021   5.465  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;1&gt;R&lt;SER&gt;A&lt;N&gt;   c&lt;solvent&gt;                     35.440  5.9    .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;2&gt;a&lt;3&gt;R&lt;SER&gt;A&lt;C&gt;         11.608  1.514  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;2&gt;a&lt;4&gt;R&lt;SER&gt;A&lt;O&gt;         0.327   2.405  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;2&gt;a&lt;5&gt;R&lt;SER&gt;A&lt;CB&gt;        14.170  1.523  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;2&gt;a&lt;6&gt;R&lt;SER&gt;A&lt;OG&gt;        0.820   2.430  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;3&gt;a&lt;7&gt;R&lt;LYS&gt;A&lt;N&gt;         3.902   2.371  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;A&gt;r&lt;5&gt;a&lt;24&gt;R&lt;GLU&gt;A&lt;CB&gt;       0.081   4.954  .  .
c&lt;A&gt;r&lt;2&gt;a&lt;2&gt;R&lt;SER&gt;A&lt;CA&gt;  c&lt;solvent&gt;                     16.448  6.1    .  .</code></pre>
<p>Each atomic descriptor in the above text can be transformed into a space-separated list (chainID resSeq iCode serial altLoc resName name) using “voronota expand-descriptors” command:</p>
<pre><code>cat annotated_contacts.txt | voronota expand-descriptors | column -t &gt; expanded_table_of_contacts.txt</code></pre>
<p>Below is a part of some possible “expanded_table_of_contacts.txt”:</p>
<pre><code>A  2  .  1  .  SER  N   A        2    .  2     .  SER  CA   15.908  1.456  .  .
A  2  .  1  .  SER  N   A        2    .  3     .  SER  C    0.167   2.488  .  .
A  2  .  1  .  SER  N   A        2    .  4     .  SER  O    7.025   2.774  .  .
A  2  .  1  .  SER  N   A        2    .  5     .  SER  CB   7.021   2.486  .  .
A  2  .  1  .  SER  N   A        2    .  6     .  SER  OG   0.624   3.159  .  .
A  2  .  1  .  SER  N   A        5    .  24    .  GLU  CB   2.849   4.628  .  .
A  2  .  1  .  SER  N   A        5    .  26    .  GLU  CD   0.008   4.792  .  .
A  2  .  1  .  SER  N   A        5    .  27    .  GLU  OE1  11.323  3.932  .  .
A  2  .  1  .  SER  N   A        194  .  1501  .  LEU  CD2  0.021   5.465  .  .
A  2  .  1  .  SER  N   solvent  .    .  .     .  .    .    35.440  5.9    .  .
A  2  .  2  .  SER  CA  A        2    .  3     .  SER  C    11.608  1.514  .  .
A  2  .  2  .  SER  CA  A        2    .  4     .  SER  O    0.327   2.405  .  .
A  2  .  2  .  SER  CA  A        2    .  5     .  SER  CB   14.170  1.523  .  .
A  2  .  2  .  SER  CA  A        2    .  6     .  SER  OG   0.820   2.430  .  .
A  2  .  2  .  SER  CA  A        3    .  7     .  LYS  N    3.902   2.371  .  .
A  2  .  2  .  SER  CA  A        5    .  24    .  GLU  CB   0.081   4.954  .  .
A  2  .  2  .  SER  CA  solvent  .    .  .     .  .    .    16.448  6.1    .  .</code></pre>
<h2 id="querying-annotated-inter-atom-contacts">Querying annotated inter-atom contacts</h2>
<p>The “voronota query-contacts” command can be used to query and manipulate computed annotated contacts. For example, below is the command that selects contacts that probably correspond to salt bridges:</p>
<pre><code>cat annotated_contacts.txt | voronota query-contacts --match-first &#39;R&lt;ASP,GLU&gt;&amp;A&lt;OD1,OD2,OE1,OE2,OXT&gt;&#39; --match-second &#39;R&lt;ARG,HIS,LYS&gt;&amp;A&lt;NH1,NH2,ND1,NE2,NZ&gt;&#39; --match-max-dist 4.0</code></pre>
<p>The <a href="support/generate-arguments-for-query-contacts.html">support/generate-arguments-for-query-contacts.html</a> file in the Voronota package provides a graphical interface for generating sets of arguments for the “voronota query-contacts” command: it serves as an interactive documentation. A similar interface for the “voronota query-balls” command is provided by the <a href="support/generate-arguments-for-query-balls.html">support/generate-arguments-for-query-balls.html</a> file.</p>
<h2 id="getting-help-in-command-line">Getting help in command line</h2>
<p>The list of all available Voronota commands is displayed when executing Voronota without any parameters.</p>
<p>Command help is shown when “–help” command line option is present, for example:</p>
<pre><code>./voronota calculate-vertices --help</code></pre>
<p>Using “–help” option without specific command results in printing help for all commands:</p>
<pre><code>./voronota --help</code></pre>
<h1 id="command-reference">Command reference</h1>
<h2 id="list-of-all-commands">List of all commands</h2>
<ul>
<li>get-balls-from-atoms-file</li>
<li>calculate-vertices</li>
<li>calculate-vertices-in-parallel</li>
<li>calculate-contacts</li>
<li>query-balls</li>
<li>query-contacts</li>
<li>draw-contacts</li>
<li>score-contacts-energy</li>
<li>score-contacts-quality</li>
<li>score-contacts-potential</li>
<li>compare-contacts</li>
<li>write-balls-to-atoms-file</li>
<li>query-balls-clashes</li>
<li>run-script</li>
<li>expand-descriptors</li>
</ul>
<h2 id="command-get-balls-from-atoms-file">Command ‘get-balls-from-atoms-file’</h2>
<h3 id="command-line-arguments">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--annotated                                 flag to enable annotated mode
--include-heteroatoms                       flag to include heteroatoms
--include-hydrogens                         flag to include hydrogen atoms
--multimodel-chains                         flag to read multiple models in PDB format and rename chains accordingly
--input-format                  string      input format, variants are: &#39;pdb&#39; (default), &#39;mmcif&#39;, &#39;detect&#39;
--radii-file                    string      path to radii configuration file
--default-radius                number      default atomic radius
--only-default-radius                       flag to make all radii equal to the default radius
--hull-offset                   number      positive offset distance enables adding artificial hull balls
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream">Input stream:</h3>
<pre><code>file in PDB or mmCIF format</code></pre>
<h3 id="output-stream">Output stream:</h3>
<pre><code>list of balls

    default mode line format: &#39;x y z r # atomSerial chainID resSeq resName atomName altLoc iCode&#39;

    annotated mode line format: &#39;annotation x y z r tags adjuncts&#39;</code></pre>
<h2 id="command-calculate-vertices">Command ‘calculate-vertices’</h2>
<h3 id="command-line-arguments-1">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--print-log                                 flag to print log of calculations
--exclude-hidden-balls                      flag to exclude hidden input balls
--include-surplus-quadruples                flag to include surplus quadruples
--link                                      flag to output links between vertices
--init-radius-for-BSH           number      initial radius for bounding sphere hierarchy
--check                                     flag to slowly check the resulting vertices (used only for testing)
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-1">Input stream:</h3>
<pre><code>list of balls (line format: &#39;x y z r&#39;)</code></pre>
<h3 id="output-stream-1">Output stream:</h3>
<pre><code>list of Voronoi vertices, i.e. quadruples with tangent spheres (line format: &#39;q1 q2 q3 q4 x y z r&#39;)</code></pre>
<h2 id="command-calculate-vertices-in-parallel">Command ‘calculate-vertices-in-parallel’</h2>
<h3 id="command-line-arguments-2">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--method                        string   *  parallelization method name, variants are: &#39;simulated&#39;
--parts                         number   *  number of parts for splitting, must be power of 2
--print-log                                 flag to print log of calculations
--include-surplus-quadruples                flag to include surplus quadruples
--link                                      flag to output links between vertices
--init-radius-for-BSH           number      initial radius for bounding sphere hierarchy
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-2">Input stream:</h3>
<pre><code>list of balls (line format: &#39;x y z r&#39;)</code></pre>
<h3 id="output-stream-2">Output stream:</h3>
<pre><code>list of Voronoi vertices, i.e. quadruples with tangent spheres (line format: &#39;q1 q2 q3 q4 x y z r&#39;)</code></pre>
<h2 id="command-calculate-contacts">Command ‘calculate-contacts’</h2>
<h3 id="command-line-arguments-3">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--annotated                                 flag to enable annotated mode
--probe                         number      probe radius
--exclude-hidden-balls                      flag to exclude hidden input balls
--step                          number      curve step length
--projections                   number      curve optimization depth
--sih-depth                     number      spherical surface optimization depth
--add-mirrored                              flag to add mirrored contacts to non-annnotated output
--draw                                      flag to output graphics for annotated contacts
--tag-centrality                            flag to tag contacts centrality
--tag-peripherial                           flag to tag peripherial contacts
--old-contacts-output           string      file path to output spherical contacts
--volumes-output                string      file path to output constrained cells volumes
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-3">Input stream:</h3>
<pre><code>list of balls

    default mode line format: &#39;x y z r&#39;

    annotated mode line format: &#39;annotation x y z r tags adjuncts&#39;</code></pre>
<h3 id="output-stream-3">Output stream:</h3>
<pre><code>list of contacts

    default mode line format: &#39;b1 b2 area&#39;

    annotated mode line format: &#39;annotation1 annotation2 area distance tags adjuncts [graphics]&#39;</code></pre>
<h2 id="command-query-balls">Command ‘query-balls’</h2>
<h3 id="command-line-arguments-4">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--match                         string      selection
--match-not                     string      negative selection
--match-tags                    string      tags to match
--match-tags-not                string      tags to not match
--match-adjuncts                string      adjuncts intervals to match
--match-adjuncts-not            string      adjuncts intervals to not match
--match-external-annotations    string      file path to input matchable annotations
--invert                                    flag to invert selection
--whole-residues                            flag to select whole residues
--drop-atom-serials                         flag to drop atom serial numbers from input
--drop-altloc-indicators                    flag to drop alternate location indicators from input
--drop-tags                                 flag to drop all tags from input
--drop-adjuncts                             flag to drop all adjuncts from input
--set-tags                      string      set tags instead of filtering
--set-dssp-info                 string      file path to input DSSP file
--set-adjuncts                  string      set adjuncts instead of filtering
--set-external-adjuncts         string      file path to input external adjuncts
--set-external-adjuncts-name    string      name for external adjuncts
--rename-chains                             flag to rename input chains to be in interval from &#39;A&#39; to &#39;Z&#39;
--guess-chain-names                         flag to assign input chain names based on residue numbering
--renumber-from-adjunct         string      adjunct name to use for input residue renumbering
--renumber-positively                       flag to increment residue numbers to make them positive
--reset-serials                             flag to reset atom serial numbers
--set-seq-pos-adjunct                       flag to set normalized sequence position adjunct
--set-ref-seq-num-adjunct       string      file path to input reference sequence
--ref-seq-alignment             string      file path to output alignment with reference
--seq-output                    string      file path to output query result sequence string
--chains-summary-output         string      file path to output chains summary
--chains-seq-identity           number      sequence identity threshold for chains summary, default is 0.9
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-4">Input stream:</h3>
<pre><code>list of balls (line format: &#39;annotation x y z r tags adjuncts&#39;)</code></pre>
<h3 id="output-stream-4">Output stream:</h3>
<pre><code>list of balls (line format: &#39;annotation x y z r tags adjuncts&#39;)</code></pre>
<h2 id="command-query-contacts">Command ‘query-contacts’</h2>
<h3 id="command-line-arguments-5">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--match-first                   string      selection for first contacting group
--match-first-not               string      negative selection for first contacting group
--match-second                  string      selection for second contacting group
--match-second-not              string      negative selection for second contacting group
--match-min-seq-sep             number      minimum residue sequence separation
--match-max-seq-sep             number      maximum residue sequence separation
--match-min-area                number      minimum contact area
--match-max-area                number      maximum contact area
--match-min-dist                number      minimum distance
--match-max-dist                number      maximum distance
--match-tags                    string      tags to match
--match-tags-not                string      tags to not match
--match-adjuncts                string      adjuncts intervals to match
--match-adjuncts-not            string      adjuncts intervals to not match
--match-external-first          string      file path to input matchable annotations
--match-external-second         string      file path to input matchable annotations
--match-external-pairs          string      file path to input matchable annotation pairs
--no-solvent                                flag to not include solvent accessible areas
--ignore-dist-for-solvent                   flag to ignore distance for solvent contacts
--ignore-seq-sep-for-solvent                flag to ignore sequence separation for solvent contacts
--no-same-chain                             flag to not include same chain contacts
--no-poly-bonds                             flag to not include peptide and nucleic polymerization bonds
--invert                                    flag to invert selection
--drop-tags                                 flag to drop all tags from input
--drop-adjuncts                             flag to drop all adjuncts from input
--set-tags                      string      set tags instead of filtering
--set-hbplus-tags               string      file path to input HBPLUS file
--set-distance-bins-tags        string      list of distance thresholds
--inter-residue-hbplus-tags                 flag to set inter-residue H-bond tags
--set-adjuncts                  string      set adjuncts instead of filtering
--set-external-adjuncts         string      file path to input external adjuncts
--set-external-adjuncts-name    string      name for external adjuncts
--set-external-means            string      file path to input external values for averaging
--set-external-means-name       string      name for external means
--renaming-map                  string      file path to input atoms renaming map
--inter-residue                             flag to convert input to inter-residue contacts
--inter-residue-after                       flag to convert output to inter-residue contacts
--summing-exceptions            string      file path to input inter-residue summing exceptions annotations
--summarize                                 flag to output only summary of matched contacts
--summarize-by-first                        flag to output only summary of matched contacts by first identifier
--preserve-graphics                         flag to preserve graphics in output
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-5">Input stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 area distance tags adjuncts [graphics]&#39;)</code></pre>
<h3 id="output-stream-5">Output stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 area distance tags adjuncts [graphics]&#39;)</code></pre>
<h2 id="command-draw-contacts">Command ‘draw-contacts’</h2>
<h3 id="command-line-arguments-6">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--drawing-for-pymol             string      file path to output drawing as pymol script
--drawing-for-jmol              string      file path to output drawing as jmol script
--drawing-for-scenejs           string      file path to output drawing as scenejs script
--drawing-for-chimera           string      file path to output drawing as chimera bild script
--drawing-name                  string      graphics object name for drawing output
--default-color                 string      default color for drawing output, in hex format, white is 0xFFFFFF
--adjunct-gradient              string      adjunct name to use for gradient-based coloring
--adjunct-gradient-blue         number      blue adjunct gradient value
--adjunct-gradient-red          number      red adjunct gradient value
--adjuncts-rgb                              flag to use RGB color values from adjuncts
--random-colors                             flag to use random color for each drawn contact
--alpha                         number      alpha opacity value for drawing output
--use-labels                                flag to use labels in drawing if possible
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-6">Input stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 area distance tags adjuncts graphics&#39;)</code></pre>
<h3 id="output-stream-6">Output stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 area distance tags adjuncts graphics&#39;)</code></pre>
<h2 id="command-score-contacts-energy">Command ‘score-contacts-energy’</h2>
<h3 id="command-line-arguments-7">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--potential-file                string   *  file path to input potential values
--ignorable-max-seq-sep         number      maximum residue sequence separation for ignorable contacts
--inter-atom-scores-file        string      file path to output inter-atom scores
--atom-scores-file              string      file path to output atom scores
--depth                         number      neighborhood normalization depth
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-7">Input stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 conditions area&#39;)</code></pre>
<h3 id="output-stream-7">Output stream:</h3>
<pre><code>global scores</code></pre>
<h2 id="command-score-contacts-quality">Command ‘score-contacts-quality’</h2>
<h3 id="command-line-arguments-8">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--default-mean                  number      default mean parameter
--default-sd                    number      default standard deviation parameter
--means-and-sds-file            string      file path to input atomic mean and sd parameters
--mean-shift                    number      mean shift in standard deviations
--external-weights-file         string      file path to input external weights for global scoring
--smoothing-window              number      window to smooth residue quality scores along sequence
--atom-scores-file              string      file path to output atom scores
--residue-scores-file           string      file path to output residue scores
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-8">Input stream:</h3>
<pre><code>list of atom energy descriptors</code></pre>
<h3 id="output-stream-8">Output stream:</h3>
<pre><code>weighted average local score</code></pre>
<h2 id="command-score-contacts-potential">Command ‘score-contacts-potential’</h2>
<h3 id="command-line-arguments-9">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--input-file-list                           flag to read file list from stdin
--input-contributions           string      file path to input contact types contributions
--input-fixed-types             string      file path to input fixed types
--input-seq-pairs-stats         string      file path to input sequence pairings statistics
--potential-file                string      file path to output potential values
--probabilities-file            string      file path to output observed and expected probabilities
--single-areas-file             string      file path to output single type total areas
--contributions-file            string      file path to output contact types contributions
--multiply-areas                number      coefficient to multiply output areas
--toggling-list                 string      list of toggling subtags
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-9">Input stream:</h3>
<pre><code>list of contacts (line format: &#39;annotation1 annotation2 conditions area&#39;)</code></pre>
<h3 id="output-stream-9">Output stream:</h3>
<pre><code>line of contact type area summaries (line format: &#39;annotation1 annotation2 conditions area&#39;)</code></pre>
<h2 id="command-compare-contacts">Command ‘compare-contacts’</h2>
<h3 id="command-line-arguments-10">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--target-contacts-file          string   *  file path to input target contacts
--inter-atom-scores-file        string      file path to output inter-atom scores
--inter-residue-scores-file     string      file path to output inter-residue scores
--atom-scores-file              string      file path to output atom scores
--residue-scores-file           string      file path to output residue scores
--depth                         number      local neighborhood depth
--smoothing-window              number      window to smooth residue scores along sequence
--smoothed-scores-file          string      file path to output smoothed residue scores
--ignore-residue-names                      flag to consider just residue numbers and ignore residue names
--residue-level-only                        flag to output only residue-level results
--detailed-output                           flag to enable detailed output
--chains-renaming-file          string      file path to input chains renaming
--remap-chains                              flag to calculate optimal chains remapping
--remap-chains-log                          flag output remapping progress to stderr
--remapped-chains-file          string      file path to output calculated chains remapping
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-10">Input stream:</h3>
<pre><code>list of model contacts (line format: &#39;annotation1 annotation2 area&#39;)</code></pre>
<h3 id="output-stream-10">Output stream:</h3>
<pre><code>global scores (atom-level and residue-level)</code></pre>
<h2 id="command-write-balls-to-atoms-file">Command ‘write-balls-to-atoms-file’</h2>
<h3 id="command-line-arguments-11">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--pdb-output                    string      file path to output query result in PDB format
--pdb-output-b-factor           string      name of adjunct to output as B-factor in PDB format
--pdb-output-template           string      file path to input template for B-factor insertions
--add-chain-terminators                     flag to add TER lines after chains
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-11">Input stream:</h3>
<pre><code>list of balls (line format: &#39;annotation x y z r tags adjuncts&#39;)</code></pre>
<h3 id="output-stream-11">Output stream:</h3>
<pre><code>list of balls (line format: &#39;annotation x y z r tags adjuncts&#39;)</code></pre>
<h2 id="command-query-balls-clashes">Command ‘query-balls-clashes’</h2>
<h3 id="command-line-arguments-12">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--clash-distance                number      clash distance threshold in angstroms, default is 3.0
--min-seq-sep                   number      minimum residue sequence separation, default is 2
--init-radius-for-BSH           number      initial radius for bounding sphere hierarchy
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-12">Input stream:</h3>
<pre><code>list of balls (line format: &#39;annotation x y z r&#39;)</code></pre>
<h3 id="output-stream-12">Output stream:</h3>
<pre><code>list of clashes (line format: &#39;annotation1 annotation2 distance min-distance-between-balls&#39;)</code></pre>
<h2 id="command-run-script">Command ‘run-script’</h2>
<h3 id="command-line-arguments-13">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--interactive                               flag for interactive mode
--exit-on-first-failure                     flag to terminate script when a command fails
--max-unfolding                 number      maximum level of output unfolding, default is 6
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-13">Input stream:</h3>
<pre><code>script as plain text</code></pre>
<h3 id="output-stream-13">Output stream:</h3>
<pre><code>output of script execution</code></pre>
<h2 id="command-expand-descriptors">Command ‘expand-descriptors’</h2>
<h3 id="command-line-arguments-14">Command line arguments:</h3>
<pre><code>Name                            Type        Description
------------------------------- ------ ---- ------------------------------------------------------------------------
--help                                      flag to print usage help to stdout and exit</code></pre>
<h3 id="input-stream-14">Input stream:</h3>
<pre><code>any text containing atom descriptors</code></pre>
<h3 id="output-stream-14">Output stream:</h3>
<pre><code>text with each atom descriptor expanded to &#39;chainID resSeq iCode serial altLoc resName name&#39;</code></pre>
<h1 id="wrapper-scripts">Wrapper scripts</h1>
<h2 id="voromqa-method-script">VoroMQA method script</h2>
<p>‘voronota-voromqa’ script is an implementation of VoroMQA (Voronoi diagram-based Model Quality Assessment) method using Voronota. The script interface is presented below:</p>
<pre><code>Basic options:
    --input | -i                   string      input structure file in PDB or mmCIF format
    --input-filter-query           string      input atoms filtering query parameters
    --output-atom-scores           string      output text file with atom scores
    --output-atom-scores-pdb       string      output PDB file with atom scores as B-factors
    --output-residue-scores        string      output text file with residue scores
    --output-residue-scores-pdb    string      output PDB file with residue scores as B-factors
    --output-residue-scores-plot   string      output PNG image file with residue scores plot, requires R
    --help | -h                                flag to display help message and exit

Advanced options:
    --cache-dir                    string      path to cache directory
    --smoothing-window             number      residue scores smoothing window size, default is 5
    --atoms-query                  string      atoms query parameters to define selection
    --contacts-query               string      contacts query parameters to define selection
    --output-selected-scores       string      output text file with selected atom scores
    --reference-sequence           string      sequence file or string for residue renumbering
    --output-sequence-alignment    string      output text file with sequence alignment
    --print-header                             flag to print output header
    --multiple-models                          flag to handle multiple models in PDB file
    --score-inter-chain                        flag to output inter-chain interface scores
    --list-all-options                         flag to display list of all command line options and exit

Standard output (one line):
    {input file path} {global score} {number of residues} {number of atoms} [ {selection score} {number of selected atoms} ]</code></pre>
<h3 id="full-list-of-options-for-the-voromqa-method-script">Full list of options for the VoroMQA method script</h3>
<p>All options of ‘voronota-voromqa’ script, in alphabetical order:</p>
<pre><code>    --add-hydrogens                            string      command to add hydrogens
    --atoms-query                              string      atoms query parameters to define selection
    --cache-dir                                string      path to cache directory
    --contacts-query                           string      contacts query parameters to define selection
    --energy-input-potential                   string      path to custom potential file
    --energy-input-statistics                  string      path to custom energy statistics file
    --energy-mode                              string      energy mode name
    --help                                                 flag to display help message and exit
    --highlight-selection-in-plot                          flag to highlight selection in plot
    --input-filter-query                       string      input atoms filtering query parameters
    --input-is-list                                        flag to treat input as a list of files
    --input-is-structure                                   flag to treat input as a PDB file
    --input                                    string      input structure file in PDB format
    --more-logging                                         flag to enable more logging
    --multiple-energy-modes                    string      list of energy modes
    --multiple-models                                      flag to handle multiple models in PDB file
    --mutation-goal                            string      three-letter residue code
    --mutation-query                           string      atoms query parameters to define residues to mutate
    --neglect-SAS                                          flag to discard solvent-accessible surface
    --no-multiple-energy-modes                             flag to not use multiple energy modes
    --no-split-into-models                                 flag to not split input into separate models
    --no-use-slurm                                         flag to not use Slurm
    --options-for-calculating-contacts         string      custom options for calculating contacts
    --output-atom-depth-values                 string      output file with atom depth values
    --output-atom-energies                     string      output file with atom energy values
    --output-atom-scores-pdb                   string      output PDB file with atom scores as B-factors
    --output-atom-scores                       string      output text file with atom scores
    --output-balls                             string      output file with balls
    --output-contacts-map-svg                  string      output file with contact map image
    --output-contacts                          string      output file with contacts
    --output-cutting-suggestions               string      output file with chain cutting suggestions
    --output-directory-for-global-scores       string      output directory for global scores
    --output-residue-scores-for-CAMEO          string      output file with output for CAMEO
    --output-residue-scores-like-for-CASP      string      output file with output like for CASP
    --output-residue-scores-pdb                string      output PDB file with residue scores as B-factors
    --output-residue-scores-plot               string      output PNG image file with residue scores plot, requires R
    --output-residue-scores                    string      output text file with residue scores
    --output-scores-for-CASP-log               string      output file with logs generated in CASP mode
    --output-scores-for-CASP-pdb               string      output file with structure generated in CASP mode
    --output-scores-for-CASP                   string      output file with scores generated in CASP mode
    --output-screenshot                        string      output file with screenshot
    --output-selected-scores                   string      output text file with selected atom scores
    --output-selected-scores-by-residue        string      output text file with selected scores summed by residue
    --output-sequence-alignment                string      output text file with sequence alignment
    --output-summary-for-potential             string      output file with statistics of contact
    --print-clash-score-of-contacts-selection              flag to print clash score of contacts selection
    --print-energy-of-contacts-selection                   flag to print energy of contacts selection
    --print-header                                         flag to print output header
    --print-horizontally                                   flag to print output horizontally
    --print-vertically                                     flag to print output vertically
    --probe-radius                             number      probe radius for defining contacts
    --processors                               number      maximum number of processors to use
    --rebuild-sidechains                       string      command to rebuild side-chains
    --reference-sequence                       string      sequence file or string for residue renumbering
    --reinterpret-SAS                                      flag to interpret solvent as hydrophobic atoms
    --sbatch-parameters                        string      parameters for sbatch when using slurm
    --score-inter-chain                                    flag to output inter-chain interface scores
    --screenshot-background                    string      screenshot background color
    --screenshot-height                        number      screenshot image height
    --screenshot-width                         number      screenshot image width
    --small-plot                                           flag to generate smaller residue scores plot
    --smoothing-window                         number      residue scores smoothing window size, default is 5
    --split-into-models                        string      path to directory for storing extracted models
    --strip-rotamers                                       flag to remove sidechains
    --surface-craving-atoms-energies-output    string      output text file with surface frustration results
    --surface-craving-atoms-energies-pdb       string      output PDB file with surface frustration results
    --surface-craving-atoms-output-pdb         string      output PDB file with atom scores as B-factors
    --surface-craving-atoms-output             string      output text file with atom scores
    --surface-craving-depth-value              number      depth for surface frustration calculation, default is 3
    --surface-craving-residues-output-pdb      string      output PDB file with residue scores as B-factors
    --surface-craving-residues-output          string      output text file with residue scores
    --w-surface-craving-residues-output-pdb    string      output PDB file with residue scores as B-factors
    --w-surface-craving-residues-output        string      output text file with residue scores
    --tag-peripheral-contacts                              flag to tag peripheral contacts
    --unteaching-dir                           string      path to directory with potential-defining files
    --use-slurm                                string      path to directory to use for Slurm logs</code></pre>
<h2 id="cad-score-method-script">CAD-score method script</h2>
<p>‘voronota-cadscore’ script is an implementation of CAD-score (Contact Area Difference score) method using Voronota. The script interface is presented below:</p>
<pre><code>Basic options:
    --input-target | -t             string   *  input target structure file in PDB or mmCIF format
    --input-model | -m              string   *  input model structure file in PDB format
    --input-filter-query            string      input atoms filtering query parameters
    --filter-model-by-target                    flag to filter model residues by the set of target residue IDs
    --output-residue-scores         string      output text file with residue scores
    --output-residue-scores-pdb-t   string      output target PDB file with residue scores as B-factors
    --output-residue-scores-pdb-m   string      output model PDB file with residue scores as B-factors
    --smoothing-window              number      residue scores smoothing window size, default is 0
    --contacts-query                string      contacts query parameters
    --contacts-query-inter-chain                flag to consider only inter-chain contacts
    --contacts-query-by-code        string      contacts query code, possible codes are AA, AS, SS, AM, MM, MS
    --use-all-query-codes                       flag to output global scores for all query codes, one line per code
    --cache-dir                     string      path to cache directory
    --output-header                             flag to output header before result line
    --help | -h                                 flag to display help message and exit

Advanced options:
    --ignore-residue-names                      flag to consider just residue numbers and ignore residue names
    --enable-site-based-scoring                 flag to enable site-based scoring
    --multiple-models                           flag to handle multiple models in PDB file as an assembly
    --input-model-chains-renaming               input text file with renaming rules for chains
    --remap-chains                              flag to automatically rearrange chain names for higher scores
    --remap-chains-output           string      output file with chain names rearrangement
    --neighborhood-depth            number      number of layers to for per-residue scoring, default is 0
    --old-regime                                flag to calculate areas as in pre-Voronota CAD-score

Standard output (one line):
    {target file path} {model file path} {query code} {number of residues} {global score} {target total area} {corresponding model total area}
    [ {site-based number of residues} {site-based global score} {site-based target total area} {corresponding site-based model total area} ]</code></pre>
<h2 id="contacts-calculation-convenience-script">Contacts calculation convenience script</h2>
<p>‘voronota-contacts’ script provides a way for calculating and querying interatomic contacts with just one command (without the need to construct a pipeline from ‘voronota’ calls).</p>
<pre><code>Basic options:
    --input | -i                   string   *  input structure file in PDB or mmCIF format
    --input-filter-query           string      input atoms filtering query parameters
    --contacts-query               string      contacts query parameters
    --contacts-query-additional    string      additional, preceeding query parameters, default is &#39;--match-min-seq-sep 1&#39;
    --cache-dir                    string      path to cache directory
    --sum-at-end                               flag to print sum of areas as the last line in output
    --tsv-output                               flag to output table in tab-separated values format with header
    --help | -h                                flag to display help message and exit

Advanced options:
    --output-drawing               string      output file with drawing script for PyMol
    --drawing-parameters           string      drawing parameters
    --wireframe-drawing                        flag to draw wireframe representation of contacts
    --multiple-models                          flag to handle multiple models in PDB file
    --use-hbplus                               flag to run &#39;hbplus&#39; to tag H-bonds

Standard output (multiple lines):
    {contacting atom} {contacting atom} {contact area} {distance between atoms centers} {tags} {adjunct values}</code></pre>
<h2 id="volumes-calculation-convenience-script">Volumes calculation convenience script</h2>
<p>‘voronota-volumes’ script provides a way for calculating and querying atomic volumes with just one command (without the need to construct a pipeline from ‘voronota’ calls).</p>
<pre><code>Basic options:
    --input | -i                   string   *  input structure file in PDB or mmCIF format
    --input-filter-query           string      input atoms filtering query parameters
    --cache-dir                    string      path to cache directory
    --sum-at-end                               flag to print sum of areas as the last line in output
    --help | -h                                flag to display help message and exit

Advanced options:
    --atoms-query                  string      atoms query parameters
    --per-residue                              flag to output per-residue results
    --multiple-models                          flag to handle multiple models in PDB file

Standard output (multiple lines):
    {name} {volume}</code></pre>
<h2 id="pocket-analysis-script">Pocket analysis script</h2>
<p>‘voronota-pocket’ script provides a way for identifying and describing pockets, cavities and channels using the Voronoi tessellation vertices.</p>
<pre><code>Options:
    --input | -i                   string   *  input structure file in PDB or mmCIF format
    --input-filter-query           string      input atoms filtering query, default is &#39;--match-tags-not het&#39;
    --probe-min                    number      scanning probe radius minimum, default is 2.0
    --probe-max                    number      scanning probe radius maximum, default is 30.0
    --buriedness-core              number      buriedness minimum for pocket start, default is 0.7
    --buriedness-rim               number      buriedness maximum for pocket end, default is 0.4
    --subpockets                   number      number of sorted subpockets to include, default is 999999
    --tangent-leeway               number      tangent sphere radius expansion for atom checks, default is 0.5
    --tangent-radius-min           number      tanget sphere radius minimum, default is 1.4
    --tetrahedron-edge-max         number      tetrahedron edge maximum, default is 999999
    --voxelization-factor          number      voxelization factor, default is 1.0
    --output-atoms                 string      file to output analyzed atoms with annotations
    --output-buriedness-pdb        string      file to output PDB file with buriedness in b-factors
    --output-pocketness-pdb        string      file to output PDB file with pocketness in b-factors
    --output-vertices              string      file to output Voronoi vertices of pocket
    --output-voxels-pdb            string      file to output voxels as PDB file
    --draw-tetrahedrons            string      file to output PyMol script for drawing pocket tetrahedrons
    --draw-spheres                 string      file to output PyMol script for drawing pocket spheres
    --output-log                   string      file to output detailed log on calculations
    --output-header                            flag to output header before result line
    --help | -h                                flag to display help message and exit

Standard output (one line):
    {input file} {max buriedness} {all atoms} {all vertices} {pocket atoms} {pocket vertices} {volume of pocket tetrahedrons}

Suggested parameters
    for pocket analysis:
        --probe-min 2 --probe-max 30 --buriedness-core 0.7 --buriedness-rim 0.4
    for large channel analysis:
        --probe-min 5 --probe-max 30 --buriedness-core 0.8 --buriedness-rim 0.6

Other suggestions:
    For large pockets or channels the spheres drawing option may
    produce very large files (too large for PyMol).</code></pre>
<h2 id="membrane-fitting-script">Membrane fitting script</h2>
<p>‘voronota-membrane’ script provides a way for fitting a membrane for a protein struture using VoroMQA-based surface frustration analysis.</p>
<pre><code>Options:
    --input | -i                   string   *  input structure file in PDB or mmCIF format
    --input-filter-query           string      input atoms filtering query, default is &#39;&#39;
    --membrane-width               number      membrane width, default is 30.0
    --output-atoms                 string      file to output analyzed atoms with annotations
    --output-membraneness-pdb      string      file to output PDB file with membraneness in b-factors
    --output-log                   string      file to output detailed log on calculations
    --output-header                            flag to output header before result line
    --help | -h                                flag to display help message and exit

Standard output (one line):
    {input file} {membrane fitting score} {direction x} {direction y} {direction z} {center projection}</code></pre>
</body>
</html>
