<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Voronota-JS version 1.29</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <style type="text/css">
  a { color: #0000CC; }
  td { padding-right: 1em; }
  pre { background-color: #DDDDDD; padding: 1em; }
  div#TOC > ul > li > ul > li ul { display: none; }
  </style>
</head>
<body>
<header id="title-block-header">
<h1 class="title">Voronota-JS version 1.29</h1>
</header>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#about-voronota-js-expansion">About Voronota-JS expansion</a>
<ul>
<li><a href="#usage-of-externally-developed-software">Usage of externally developed software</a></li>
</ul></li>
<li><a href="#quick-install-guide">Quick install guide</a></li>
<li><a href="#getting-the-latest-version">Getting the latest version</a></li>
<li><a href="#building-from-source-code">Building from source code</a>
<ul>
<li><a href="#requirements">Requirements</a></li>
<li><a href="#using-cmake">Using CMake</a></li>
<li><a href="#using-c-compiler-directly">Using C++ compiler directly</a></li>
</ul></li>
<li><a href="#more-about-protein-ligand-variation-of-cad-score">More about protein-ligand variation of CAD-score</a>
<ul>
<li><a href="#example-of-how-it-works-for-casp15-models">Example of how it works for CASP15 models</a></li>
<li><a href="#more-info-about-the-involved-contacts">More info about the involved contacts</a></li>
</ul></li>
<li><a href="#wrapper-scripts">Wrapper scripts</a>
<ul>
<li><a href="#voromqa-dark-and-light-methods">VoroMQA dark and light methods</a></li>
<li><a href="#voromqa-for-only-global-scores-with-fast-caching">VoroMQA for only global scores with fast caching</a></li>
<li><a href="#voromqa-based-membrane-protein-structure-assessment">VoroMQA-based membrane protein structure assessment</a></li>
<li><a href="#voromqa-based-collection-of-protein-protein-complex-features">VoroMQA-based collection of protein-protein complex features</a></li>
<li><a href="#fast-inter-chain-interface-voromqa-energy">Fast inter-chain interface VoroMQA energy</a></li>
<li><a href="#fast-inter-chain-interface-cad-score">Fast inter-chain interface CAD-score</a></li>
<li><a href="#fast-inter-chain-interface-cad-score-matrix">Fast inter-chain interface CAD-score matrix</a></li>
<li><a href="#fast-inter-chain-interface-contacts">Fast inter-chain interface contacts</a></li>
<li><a href="#computation-of-inter-chain-interface-graphs">Computation of inter-chain interface graphs</a></li>
<li><a href="#voroif-gnn-method-for-scoring-models-of-protein-protein-complexes">VoroIF-GNN method for scoring models of protein-protein complexes</a></li>
<li><a href="#protein-ligand-interface-variation-of-cad-score">Protein-ligand interface variation of CAD-score</a></li>
</ul></li>
</ul>
</nav>
<h1 id="about-voronota-js-expansion">About Voronota-JS expansion</h1>
<p>Voronota-JS is an expansion of <a href="../index.html">the core Voronota software</a>. Voronota-JS provides a way to write JavaScript scripts for the comprehensive analysis of macromolecular structures, including the Voronoi tesselation-based analysis.</p>
<p>Currently, the Voronota-JS package contains several executables:</p>
<ul>
<li>“voronota-js” - core engine that executes JavaScript scripts.</li>
<li>“voronota-js-voromqa” - wrapper to a voronota-js program for computing VoroMQA scores, both old and new (developed for CASP14).</li>
<li>“voronota-js-only-global-voromqa” - wrapper to a voronota-js program for computing only global VoroMQA scores with fast caching.</li>
<li>“voronota-js-membrane-voromqa” - wrapper to a voronota-js program for the VoroMQA-based analysis and assessment of membrane protein structural models.</li>
<li>“voronota-js-ifeatures-voromqa” - wrapper to a voronota-js program for the computation of multiple VoroMQA-based features of protein-protein complexes.</li>
<li>“voronota-js-fast-iface-voromqa” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface VoroMQA energy.</li>
<li>“voronota-js-fast-iface-cadscore” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface CAD-score.</li>
<li>“voronota-js-fast-iface-cadscore-matrix” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface CAD-score matrix.</li>
<li>“voronota-js-fast-iface-contacts” - wrapper to a voronota-js program for the very fast computation of the inter-chain interface contact areas.</li>
<li>“voronota-js-fast-iface-data-graph” - wrapper to a voronota-js program for the computation of interface graphs used by the VoroIF-GNN method.</li>
<li>“voronota-js-voroif-gnn” - wrapper to a voronota-js program and GNN inference scripts that run the VoroIF-GNN method for scoring models of protein-protein complexes (developed for CASP15).</li>
<li>“voronota-js-ligand-cadscore” - wrapper to a voronota-js program for the computation of protein-ligand variation of CAD-score (developed to analyze protein-ligand models from CASP15).</li>
</ul>
<h2 id="usage-of-externally-developed-software">Usage of externally developed software</h2>
<p>Voronota-JS relies on several externally developed software projects (big thanks to their authors):</p>
<ul>
<li>Duktape - <a href="https://duktape.org/">https://duktape.org/</a></li>
<li>PStreams - <a href="https://github.com/jwakely/pstreams">https://github.com/jwakely/pstreams</a></li>
<li>TM-align - <a href="https://zhanggroup.org/TM-align/">https://zhanggroup.org/TM-align/</a></li>
<li>FASPR - <a href="https://zhanggroup.org/FASPR/">https://zhanggroup.org/FASPR/</a></li>
<li>QCP - <a href="https://theobald.brandeis.edu/qcp/">https://theobald.brandeis.edu/qcp/</a></li>
<li>linenoise - <a href="https://github.com/antirez/linenoise">https://github.com/antirez/linenoise</a></li>
<li>PicoSHA2 - <a href="https://github.com/okdshin/PicoSHA2">https://github.com/okdshin/PicoSHA2</a></li>
<li>frugally-deep - <a href="https://github.com/Dobiasd/frugally-deep">https://github.com/Dobiasd/frugally-deep</a>
<ul>
<li>FunctionalPlus - <a href="https://github.com/Dobiasd/FunctionalPlus">https://github.com/Dobiasd/FunctionalPlus</a></li>
<li>Eigen - <a href="http://eigen.tuxfamily.org/">http://eigen.tuxfamily.org/</a></li>
<li>json - <a href="https://github.com/nlohmann/json">https://github.com/nlohmann/json</a></li>
</ul></li>
<li>tinf - <a href="https://github.com/jibsen/tinf">https://github.com/jibsen/tinf</a></li>
<li>LodePNG - <a href="https://github.com/lvandeve/lodepng">https://github.com/lvandeve/lodepng</a></li>
</ul>
<h1 id="quick-install-guide">Quick install guide</h1>
<p>Please refer to the <a href="../index.html#quick-install-guide">core Voronota quick install guide</a>.</p>
<h1 id="getting-the-latest-version">Getting the latest version</h1>
<p>Download the latest archive from the official downloads page: <a href="https://github.com/kliment-olechnovic/voronota/releases">https://github.com/kliment-olechnovic/voronota/releases</a>.</p>
<p>The archive contains the Voronota-JS software in the ‘expansion_js’ subdirectory. There is a ready-to-use statically compiled ‘voronota-js’ program for 64 bit Linux systems. This executable can be rebuilt from the provided source code to work on any modern Linux, macOS or Windows operating systems.</p>
<h1 id="building-from-source-code">Building from source code</h1>
<h2 id="requirements">Requirements</h2>
<p>Voronota-JS has no required external dependencies, only a C++14-compliant compiler is needed to build it.</p>
<h2 id="using-cmake">Using CMake</h2>
<p>You can build using CMake for makefile generation. Starting in the directory containing “CMakeLists.txt” file, run the sequence of commands:</p>
<pre><code>cmake ./
make</code></pre>
<p>Alternatively, to keep files more organized, CMake can be run in a separate “build” directory:</p>
<pre><code>mkdir build
cd build
cmake ../
make
cp ./voronota-js ../voronota-js</code></pre>
<h2 id="using-c-compiler-directly">Using C++ compiler directly</h2>
<p>For example, “voronota-js” executable can be built from the sources in “src” directory using GNU C++ compiler:</p>
<pre><code>g++ -std=c++14 -I&quot;./src/dependencies&quot; -O3 -o &quot;./voronota-js&quot; $(find ./src/ -name &#39;*.cpp&#39;)</code></pre>
<h1 id="more-about-protein-ligand-variation-of-cad-score">More about protein-ligand variation of CAD-score</h1>
<p>The script ‘voronota-js-ligand-cadscore’ for the protein-ligand variation of CAD-score is included to <a href="./index.html">the Voronota-JS expansion</a> of Voronota.</p>
<h2 id="example-of-how-it-works-for-casp15-models">Example of how it works for CASP15 models</h2>
<p>The ‘voronota-js-ligand-cadscore’ scripts has a special input mode for CASP15 protein-ligand model files: when using –casp15-* options for specifying input, the input files are split into receptor and ligand parts and passed to the same script with basic input options. The target input in the CASP15 mode should be formatted in the same as the model input. The basic input mode option is more flexible, but it requires all the receptor and ligand files available separately, and the ligand IDs (names) to be specified explicitly.</p>
<p>For example, let us take two files with protein-ligand models from CASP15: ‘H1135LG035_1’ and ‘H1135LG046_1’. Then let us execute the following command:</p>
<pre><code>voronota-js-ligand-cadscore \
  --casp15-target ./T1118v1LG035_1 \
  --casp15-model ./T1118v1LG046_1 \
  --casp15-target-pose 1 \
  --casp15-model-pose 2 \
| column -t</code></pre>
<p>which outputs:</p>
<pre><code>target                model                 contacts_set             CAD_score  target_area  model_area
T1118v1LG035_1_pose1  T1118v1LG046_1_pose2  interface                0.568301   648.036      808.139
T1118v1LG035_1_pose1  T1118v1LG046_1_pose2  interface_plus_adjacent  0.629109   1781.9       2189.88</code></pre>
<p>Note, that CAD-score values will be different if we swap target and model inputs. The script can do the swap automatically and output additional results when the ‘–and-swap true’ is added to the arguments, then the output will look as follows:</p>
<pre><code>target                model                 contacts_set             CAD_score  target_area  model_area
T1118v1LG035_1_pose1  T1118v1LG046_1_pose2  interface                0.568301   648.036      808.139
T1118v1LG035_1_pose1  T1118v1LG046_1_pose2  interface_plus_adjacent  0.629109   1781.9       2189.88
T1118v1LG046_1_pose2  T1118v1LG035_1_pose1  interface                0.528076   808.139      648.036
T1118v1LG046_1_pose2  T1118v1LG035_1_pose1  interface_plus_adjacent  0.519405   2189.88      1781.9</code></pre>
<h3 id="output-explanation">Output explanation</h3>
<p>The output table columns mean the following</p>
<ul>
<li>‘target’ - name of the target protein-ligand complex structure.</li>
<li>‘model’ - name of the model protein-ligand complex structure.</li>
<li>‘contacts_set’ - indicates what contacts were compared: only the direct interface contacts (‘interface’), or the expanded set that also contains the contacts adjacent to the interface (‘interface_plus_adjacent’).</li>
<li>‘CAD_score’ - CAD-score value computed using the standard CAD-score formula (the formula can be seen on <a href="https://bioinformatics.lt/cad-score/theory">https://bioinformatics.lt/cad-score/theory</a>).</li>
<li>‘target_area’ - total area of the relevant contacts in the target complex</li>
<li>‘model_area’ - total area of the relevant contacts in the model complex</li>
</ul>
<h2 id="more-info-about-the-involved-contacts">More info about the involved contacts</h2>
<p>There are some important aspects:</p>
<ul>
<li>All the contact areas are computed on the atom-atom level, but then bundled (summed) to the residue-residue/residue-ligand level.</li>
<li>When defining the contact uniqueness, residue names and numbers (and chain ID, if it is present) are used for the protein side, but only ligand names are used for the ligand side. Therefore, the script treats same-named ligands as a single ligand when summarizing contact areas. This was done to avoid doing permutations of ligand numbers for identical ligands.</li>
<li>The ‘interface’ set includes only the protein-ligand contacts.</li>
<li>The ‘interface_plus_adjacent’ set includes the following contacts: all the ‘interface’ set contacts; protein-protein contacts between residues that have contacts with ligands.</li>
<li>You can use the ‘–details-dir’ option to look at the used contacts lists.</li>
<li>You can use the ‘–drawing-dir’ option to generate PDB files (with interface participants) and contact drawing scripts for PyMol.</li>
</ul>
<h3 id="details-example">Details example</h3>
<p>Examples of files generated using the ‘–details-dir’ option:</p>
<ul>
<li>‘interface’ contacts:
<ul>
<li><a href="./tests/jobs_output/ligand_cadscore_script/details/T1118v1LG035_1_pose1_interface_contacts.txt">T1118v1LG035_1_pose1_interface_contacts.txt</a></li>
<li><a href="./tests/jobs_output/ligand_cadscore_script/details/T1118v1LG046_1_pose2_interface_contacts.txt">T1118v1LG046_1_pose2_interface_contacts.txt</a></li>
</ul></li>
<li>‘interface_plus_adjacent’ contacts:
<ul>
<li><a href="./tests/jobs_output/ligand_cadscore_script/details/T1118v1LG035_1_pose1_interface_and_adjacent_contacts.txt">T1118v1LG035_1_pose1_interface_and_adjacent_contacts.txt</a></li>
<li><a href="./tests/jobs_output/ligand_cadscore_script/details/T1118v1LG046_1_pose2_interface_and_adjacent_contacts.txt">T1118v1LG046_1_pose2_interface_and_adjacent_contacts.txt</a></li>
</ul></li>
</ul>
<h3 id="drawing-example">Drawing example</h3>
<p>Example pymol call for displaying files generated using the ‘–drawing-dir’ option:</p>
<pre><code>pymol \
  draw_T1118v1LG035_1_pose1_interface_and_adjacent_contacts_in_pymol.py \
  T1118v1LG035_1_pose1_cutout_interface_residues_ligand.pdb \
  T1118v1LG035_1_pose1_cutout_interface_residues_receptor.pdb</code></pre>
<p>Example of visualized contacts (with direct interface contacts in green, adjacent contacts in yellow, ligand atoms in red, receptor atoms in cyan): <a href="./resources/texts/protein_ligand_3D.png">protein_ligand_3D.png</a>.</p>
<h1 id="wrapper-scripts">Wrapper scripts</h1>
<h2 id="voromqa-dark-and-light-methods">VoroMQA dark and light methods</h2>
<p>‘voronota-js-voromqa’ script provides an interface to VoroMQA dark (newer) and light (classic) methods.</p>
<h3 id="script-interface">Script interface</h3>
<pre><code>Options:
    --input | -i              string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --sequence-file           string     input sequence file
    --select-contacts         string     query to select contacts, default is &#39;[-min-seq-sep 2]&#39;
    --inter-chain             string     set query to select contacts to &#39;[-inter-chain]&#39;
    --constraints-file        string     input distance constraints file
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --output-dark-scores      string     output PDB file with dark scores as B-factors
    --output-light-scores     string     output PDB file with light scores as B-factors
    --output-alignment        string     output alignment sequence alignment file
    --cache-dir               string     path to cache directory
    --tour-sort               string     tournament sorting mode, default is &#39;_mono&#39;, options are &#39;_homo&#39;, &#39;_hetero&#39; or custom
    --sbatch-parameters       string     parameters to run Slurm sbatch if input is &#39;_list&#39;
    --split-models-prefix     string     name prefix for splitting input PDB file by models
    --smoothing-window        number     residue scores smoothing window size, default is 0
    --processors              number     maximum number of processors to use, default is 1
    --casp-b-factors                     flag to write CASP-required B-factors in output PDB files
    --use-scwrl                          flag to use Scwrl4 to rebuild side-chains
    --guess-chain-names                  flag to guess chain names based on sequence nubmering
    --order-by-residue-id                flag to order atoms by residue ids
    --input-is-script                    flag to treat input file as vs script
    --as-assembly                        flag to treat input file as biological assembly
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-voromqa --input model.pdb
    
    voronota-js-voromqa --cache-dir ./cache --input model.pdb
    
    ls *.pdb | voronota-js-voromqa --cache-dir ./cache --input _list
    
    ls *.pdb | voronota-js-voromqa --cache-dir ./cache --input _list | column -t
    
    ls *.pdb | voronota-js-voromqa --cache-dir ./cache --input _list \
      --processors 8 \
      --inter-chain \
      --tour-sort _hetero
      
    ls *.pdb | voronota-js-voromqa --cache-dir ./cache --input _list \
      --processors 8 \
      --select-contacts &#39;[-a1 [-chain A -rnum 1:500] -a2 [-chain B -rnum 1:500]]&#39; \
      --tour-sort &#39;-columns full_dark_score sel_energy -multipliers 1 -1 -tolerances 0.02 0.0&#39;</code></pre>
<h2 id="voromqa-for-only-global-scores-with-fast-caching">VoroMQA for only global scores with fast caching</h2>
<p>‘voronota-js-only-global-voromqa’ script computes global VoroMQA scores and can use fast caching.</p>
<h3 id="script-interface-1">Script interface</h3>
<pre><code>Options:
    --input | -i              string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --output-dark-pdb         string     output path for PDB file with VoroMQA-dark scores, default is &#39;&#39;
    --output-light-pdb        string     output path for PDB file with VoroMQA-light scores, default is &#39;&#39;
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --cache-dir               string     cache directory path to store results of past calls
    --run-faspr               string     path to FASPR binary to rebuild side-chains
    --input-is-script                    flag to treat input file as vs script
    --as-assembly                        flag to treat input file as biological assembly
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-only-global-voromqa --input model.pdb
    
    ls *.pdb | voronota-js-only-global-voromqa --input _list --processors 8 | column -t</code></pre>
<h2 id="voromqa-based-membrane-protein-structure-assessment">VoroMQA-based membrane protein structure assessment</h2>
<p>‘voronota-js-membrane-voromqa’ script provides an interface to the VoroMQA-based method for assessing membrane protein structures.</p>
<h3 id="script-interface-2">Script interface</h3>
<pre><code>Options:
    --input | -i               string  *  input file path
    --restrict-input           string     query to restrict input atoms, default is &#39;[]&#39;
    --membrane-width           number     membrane width or list of widths to use, default is 25.0
    --output-local-scores      string     prefix to output PDB files with local scores as B-factors
    --as-assembly                         flag to treat input file as biological assembly
    --help | -h                           flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-membrane-voromqa --input model.pdb --membrane-width 30.0
    
    voronota-js-membrane-voromqa --input model.pdb --membrane-width 20.0,25.0,30.0
    
    voronota-js-membrane-voromqa --input model.pdb \
      --membrane-width 20,25,30 \
      --output-local-scores ./local_scores/</code></pre>
<h2 id="voromqa-based-collection-of-protein-protein-complex-features">VoroMQA-based collection of protein-protein complex features</h2>
<p>‘voronota-js-ifeatures-voromqa’ script computes multiple VoroMQA-based features of protein-protein complexes.</p>
<h3 id="script-interface-3">Script interface</h3>
<pre><code>Options:
    --input | -i              string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --processors              number     maximum number of processors to use, default is 1
    --use-scwrl                          flag to use Scwrl4 to rebuild side-chains
    --as-assembly                        flag to treat input file as biological assembly
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-ifeatures-voromqa --input model.pdb
    
    ls *.pdb | voronota-js-ifeatures-voromqa --input _list --processors 8 | column -t</code></pre>
<h2 id="fast-inter-chain-interface-voromqa-energy">Fast inter-chain interface VoroMQA energy</h2>
<p>‘voronota-js-fast-iface-voromqa’ script rapidly computes VoroMQA-based interface energy of protein complexes.</p>
<h3 id="script-interface-4">Script interface</h3>
<pre><code>Options:
    --input | -i                string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input            string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts        string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --constraints-required      string     query to check required contacts, default is &#39;&#39;
    --constraints-banned        string     query to check banned contacts, default is &#39;&#39;
    --constraint-clashes        number     max allowed clash score, default is 0.9
    --output-table-file         string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --output-ia-contacts-file   string     output inter-atom contacts file path, default is &#39;&#39;
    --output-ir-contacts-file   string     output inter-residue contacts file path, default is &#39;&#39;
    --processors                number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters         string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file                string     input file path to replace stdin
    --run-faspr                 string     path to FASPR binary to rebuild side-chains
    --input-is-script                      flag to treat input file as vs script
    --as-assembly                          flag to treat input file as biological assembly
    --detailed-times                       flag to output detailed times
    --score-symmetry                       flag to score interface symmetry
    --blanket                              flag to keep nucleic acids and use blanket potential
    --help | -h                            flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-fast-iface-voromqa --input model.pdb
    
    ls *.pdb | voronota-js-fast-iface-voromqa --input _list --processors 8 | column -t</code></pre>
<h2 id="fast-inter-chain-interface-cad-score">Fast inter-chain interface CAD-score</h2>
<p>‘voronota-js-fast-iface-cadscore’ script rapidly computes interface CAD-score for two protein complexes.</p>
<h3 id="script-interface-5">Script interface</h3>
<pre><code>Options:
    --target | -t             string  *  target file path
    --model | -m              string  *  model file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts      string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --subselect-site          string     query to subselect atoms for binding site scoring, default is &#39;[]&#39;
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file              string     input file path to replace stdin
    --run-faspr               string     path to FASPR binary to rebuild side-chains
    --permuting-allowance     number     maximum number of chains for exhaustive remapping, default is 4
    --as-assembly                        flag to treat input files as biological assemblies
    --remap-chains                       flag to calculate and use optimal chains remapping
    --remap-chains-logging               flag to print log of chains remapping to stderr
    --ignore-residue-names               flag to ignore residue names in residue identifiers
    --test-common-ids                    flag to fail quickly if there are no common residues
    --crude                              flag to enable very crude faster mode
    --lt                                 flag to enable voronota-lt faster mode
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-fast-iface-cadscore --target target.pdb --model model.pdb
    
    ls *.pdb | voronota-js-fast-iface-cadscore --target target.pdb --model _list --processors 8 | column -t</code></pre>
<h2 id="fast-inter-chain-interface-cad-score-matrix">Fast inter-chain interface CAD-score matrix</h2>
<p>‘voronota-js-fast-iface-cadscore-matrix’ script rapidly computes interface CAD-score between complexes.</p>
<h3 id="script-interface-6">Script interface</h3>
<pre><code>Options:
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts      string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file              string     input file path to replace stdin
    --permuting-allowance     number     maximum number of chains for exhaustive remapping, default is 4
    --as-assembly                        flag to treat input files as biological assemblies
    --remap-chains                       flag to calculate and use optimal chains remapping
    --ignore-residue-names               flag to ignore residue names in residue identifiers
    --crude                              flag to enable very crude faster mode
    --lt                                 flag to enable voronota-lt faster mode
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of scores
    
Examples:

    ls *.pdb | voronota-js-fast-iface-cadscore-matrix | column -t
    
    find ./complexes/ -type f -name &#39;*.pdb&#39; | voronota-js-fast-iface-cadscore-matrix &gt; &quot;full_matrix.txt&quot;
    
    (find ./group1/ -type f | awk &#39;{print $1 &quot; a&quot;}&#39; ; find ./group2/ -type f | awk &#39;{print $1 &quot; b&quot;}&#39;) | voronota-js-fast-iface-cadscore-matrix &gt; &quot;itergroup_matrix.txt&quot;</code></pre>
<h2 id="fast-inter-chain-interface-contacts">Fast inter-chain interface contacts</h2>
<p>‘voronota-js-fast-iface-contacts’ script rapidly computes contacts of inter-chain interface in a molecular complex.</p>
<h3 id="script-interface-7">Script interface</h3>
<pre><code>Options:
    --input                   string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts      string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --output-contacts-file    string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --output-bsite-file       string     output binding site table file path, default is &#39;&#39;
    --output-drawing-script   string     output PyMol drawing script file path, default is &#39;&#39;
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file              string     input file path to replace stdin
    --run-faspr               string     path to FASPR binary to rebuild side-chains
    --custom-radii-file       string     path to file with van der Waals radii assignment rules
    --with-sas-areas                     flag to also compute and output solvent-accessible areas of interface residue atoms
    --coarse-grained                     flag to output inter-residue contacts
    --input-is-script                    flag to treat input file as vs script
    --as-assembly                        flag to treat input file as biological assembly
    --use-hbplus                         flag to run &#39;hbplus&#39; to tag H-bonds
    --expand-ids                         flag to output expanded IDs
    --og-pipeable                        flag to format output to be pipeable to &#39;voronota query-contacts&#39;
    --help | -h                          flag to display help message and exit

Standard output:
    tab-separated table of contacts
    
Examples:

    voronota-js-fast-iface-contacts --input &quot;./model.pdb&quot; --expand-ids &gt; &quot;./contacts.tsv&quot;
    
    voronota-js-fast-iface-contacts --input &quot;./model.pdb&quot; --with-sas-areas --coarse-grained --og-pipeable | voronota query-contacts --summarize-by-first
    
    cat &quot;./model.pdb&quot; | voronota-js-fast-iface-contacts --input _stream --with-sas-areas --coarse-grained --og-pipeable | voronota query-contacts --summarize
    
    ls *.pdb | voronota-js-fast-iface-contacts --input _list --processors 8 --output-contacts-file &quot;./output/-BASENAME-.tsv&quot;</code></pre>
<h2 id="computation-of-inter-chain-interface-graphs">Computation of inter-chain interface graphs</h2>
<p>‘voronota-js-fast-iface-data-graph’ script generates interface data graphs of protein complexes.</p>
<h3 id="script-interface-8">Script interface</h3>
<pre><code>Options:
    --input                   string  *  input file path or &#39;_list&#39; to read file paths from stdin
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts      string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --with-reference          string     input reference complex structure file path, default is &#39;&#39;
    --output-data-prefix      string  *  output data files prefix
    --output-table-file       string     output table file path, default is &#39;_stdout&#39; to print to stdout
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file              string     input file path to replace stdin
    --run-faspr               string     path to FASPR binary to rebuild side-chains
    --coarse-grained                     flag to output a coarse-grained graph
    --input-is-script                    flag to treat input file as vs script
    --as-assembly                        flag to treat input file as biological assembly
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of generated file paths
    
Examples:

    voronota-js-fast-iface-data-graph --input model.pdb --output-prefix ./data_graphs/
    
    ls *.pdb | voronota-js-fast-iface-data-graph --input _list --processors 8 --output-prefix ./data_graphs/ | column -t</code></pre>
<h2 id="voroif-gnn-method-for-scoring-models-of-protein-protein-complexes">VoroIF-GNN method for scoring models of protein-protein complexes</h2>
<p>‘voronota-js-voroif-gnn’ scores protein-protein interfaces using the VoroIF-GNN method</p>
<h3 id="script-interface-9">Script interface</h3>
<pre><code>Options:
    --input                   string  *  input file path, or &#39;_list&#39; to read multiple input files paths from stdin
    --gnn                     string     GNN package file or directory with package files, default is &#39;${SCRIPT_DIRECTORY}/voroif/gnn_packages/v1&#39;
    --gnn-add                 string     additional GNN package file or directory with package files, default is &#39;&#39;
    --restrict-input          string     query to restrict input atoms, default is &#39;[]&#39;
    --subselect-contacts      string     query to subselect inter-chain contacts, default is &#39;[]&#39;
    --as-assembly             string     flag to treat input file as biological assembly
    --input-is-script         string     flag to treat input file as vs script
    --conda-path              string     conda installation path, default is &#39;&#39;
    --conda-env               string     conda environment name, default is &#39;&#39;
    --faspr-path              string     path to FASPR binary, default is &#39;&#39;
    --run-faspr               string     flag to rebuild sidechains using FASPR, default is &#39;false&#39;
    --processors              number     maximum number of processors to run in parallel, default is 1
    --sbatch-parameters       string     sbatch parameters to run in parallel, default is &#39;&#39;
    --stdin-file              string     input file path to replace stdin, default is &#39;_stream&#39;
    --local-column            string     flag to add per-residue scores to the global output table, default is &#39;false&#39;
    --cache-dir               string     cache directory path to store results of past calls, default is &#39;&#39;
    --output-dir              string     output directory path for all results, default is &#39;&#39;
    --output-pdb-file         string     output path for PDB file with interface residue scores, default is &#39;&#39;
    --output-pdb-mode         string     mode to write b-factors (&#39;overwrite_all&#39;, &#39;overwrite_iface&#39; or &#39;combine&#39;), default is &#39;overwrite_all&#39;
    --help | -h                          flag to display help message and exit

Standard output:
    space-separated table of global scores
    
Important note about output interpretation:
    higher GNN scores are better, lower GNN scores are worse (with VoroMQA energy it is the other way around)

Examples:

    voronota-js-voroif-gnn --conda-path ~/miniconda3 --input &#39;./model.pdb&#39;
    
    voronota-js-voroif-gnn --input &#39;./model.pdb&#39; --gnn &quot;${HOME}/git/voronota/expansion_js/voroif/gnn_packages/v1&quot;
    
    find ./models/ -type f | voronota-js-voroif-gnn --conda-path ~/miniconda3 --input _list

Requirements installation example using Miniconda (may need more than 10 GB of disk space):

    # prepare Miniconda
    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
    bash Miniconda3-latest-Linux-x86_64.sh
    source ~/miniconda3/bin/activate
    # install PyTorch using instructions from &#39;https://pytorch.org/get-started/locally/&#39;
    conda install pytorch torchvision torchaudio pytorch-cuda=12.1 -c pytorch -c nvidia
    # install PyTorch Geometric using instructions from &#39;https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html&#39;
    pip install torch_geometric
    pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-2.1.0+cu121.html
    # install Pandas
    conda install pandas
    # if you do not have R installed in you system, install it (not necessarily using conda, e.g &#39;sudo apt-get install r-base&#39; in Ubuntu)
    conda install r -c conda-forge</code></pre>
<h2 id="protein-ligand-interface-variation-of-cad-score">Protein-ligand interface variation of CAD-score</h2>
<p>‘voronota-js-ligand-cadscore’ script computes protein-ligand variation of CAD-score.</p>
<h3 id="script-interface-10">Script interface</h3>
<pre><code>Input options, basic:
    --target-receptor             string  *  target receptor file path
    --target-ligands              string  *  list of target ligand file paths
    --target-ligand-ids           string  *  list of target ligand IDs
    --model-receptor              string  *  model receptor file path
    --model-ligands               string  *  list of model ligand file paths
    --model-ligand-ids            string  *  list of model ligand IDs
    --target-name                 string     target name to use for output, default is &#39;target_complex&#39;
    --model-name                  string     model name to use for output, default is &#39;model_complex&#39;

Input options, alternative:
    --casp15-target               string  *  target data file in CASP15 format, alternative to --target-* options
    --casp15-target-pose          string  *  pose number to select from the target data file in CASP15 format
    --casp15-model                string  *  model data file in CASP15 format, alternative to --model-* options
    --casp15-model-pose           string  *  pose number to select from the model data file in CASP15 format

Other options:
    --table-dir                   string     directory to output scores table file named &#39;${TARGET_NAME}_vs_${MODEL_NAME}.txt&#39;
    --details-dir                 string     directory to output lists of contacts used for scoring
    --drawing-dir                 string     directory to output files to visualize with pymol
    --and-swap                    string     flag to compute everything after swapping target and model, default is &#39;false&#39;
    --ignore-ligand-headers       string     flag to ignore title header in ligand files
    --help | -h                              display help message and exit
    
Standard output:
    space-separated table of scores
    
Examples:

    voronota-js-ligand-cadscore --casp15-target ./T1118v1LG035_1 --casp15-target-pose 1 --casp15-model ./T1118v1LG046_1 --casp15-model-pose 1
    
    voronota-js-ligand-cadscore \
      --target-receptor ./t_protein.pdb --target-ligands &#39;./t_ligand1.mol ./t_ligand2.mol ./t_ligand3.mol&#39; --target-ligand-ids &#39;a a b&#39; \
      --model-receptor ./m_protein.pdb --model-ligands &#39;./m_ligand1.mol ./m_ligand2.mol ./m_ligand3.mol&#39; --model-ligand-ids &#39;a a b&#39;</code></pre>
</body>
</html>
