<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>CAD-score-LT version 0.8</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    /* The extra [class] is a hack that increases specificity enough to
       override a similar rule in reveal.js */
    ul.task-list[class]{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      font-size: inherit;
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { color: #008000; } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { color: #008000; font-weight: bold; } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <style type="text/css">
  a { color: #0000CC; }
  td { padding-right: 1em; }
  pre { background-color: #DDDDDD; padding: 1em; }
  div#TOC > ul > li > ul > li ul { display: none; }
  </style>
  <script>window.MathJax = { tex: { inlineMath: [['$', '$'], ['\\(', '\\)']] }, options: { skipHtmlTags: ['script','noscript','style','textarea','pre','code'] } };</script>
  <script defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js"></script>
</head>
<body>
<header id="title-block-header">
<h1 class="title">CAD-score-LT version 0.8</h1>
</header>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#introduction-to-cad-score-lt" id="toc-introduction-to-cad-score-lt">Introduction to CAD-score-LT</a>
<ul>
<li><a href="#cad-score-definition-for-residue-residue-contacts" id="toc-cad-score-definition-for-residue-residue-contacts">CAD-score definition for residue-residue contacts</a></li>
<li><a href="#cad-score-definition-for-other-types-of-areas" id="toc-cad-score-definition-for-other-types-of-areas">CAD-score definition for other types of areas</a></li>
</ul></li>
<li><a href="#quickest-install-guide" id="toc-quickest-install-guide">Quickest install guide</a></li>
<li><a href="#getting-the-latest-version" id="toc-getting-the-latest-version">Getting the latest version</a></li>
<li><a href="#building-the-command-line-tool-from-source-code" id="toc-building-the-command-line-tool-from-source-code">Building the command-line tool from source code</a>
<ul>
<li><a href="#requirements" id="toc-requirements">Requirements</a></li>
<li><a href="#using-cmake" id="toc-using-cmake">Using CMake</a></li>
<li><a href="#using-c-compiler-directly" id="toc-using-c-compiler-directly">Using C++ compiler directly</a></li>
</ul></li>
<li><a href="#basic-usage" id="toc-basic-usage">Basic usage</a>
<ul>
<li><a href="#listing-command-line-options" id="toc-listing-command-line-options">Listing command-line options</a></li>
<li><a href="#choosing-scoring-types-and-scoring-levels" id="toc-choosing-scoring-types-and-scoring-levels">Choosing scoring types and scoring levels</a></li>
<li><a href="#providing-inputs" id="toc-providing-inputs">Providing inputs</a></li>
<li><a href="#outputting-a-global-scores-table" id="toc-outputting-a-global-scores-table">Outputting a global scores table</a></li>
<li><a href="#restricting-what-to-score" id="toc-restricting-what-to-score">Restricting what to score</a></li>
<li><a href="#local-scoring-outputs" id="toc-local-scoring-outputs">Local scoring outputs</a></li>
<li><a href="#chain-remapping-for-complexes" id="toc-chain-remapping-for-complexes">Chain remapping for complexes</a></li>
</ul></li>
<li><a href="#cad-score-lt-python-bindings" id="toc-cad-score-lt-python-bindings">CAD-score-LT Python bindings</a>
<ul>
<li><a href="#installation" id="toc-installation">Installation</a></li>
<li><a href="#usage-examples" id="toc-usage-examples">Usage examples</a></li>
</ul></li>
</ul>
</nav>
<h1 id="introduction-to-cad-score-lt">Introduction to CAD-score-LT</h1>
<p>CAD-score-LT software computes CAD-score (Contact Area Difference score): a superposition-free similarity measure based on contact areas. CAD-score-LT is based on the <a href="../expansion_lt/index.html">Voronota-LT software</a>. CAD-score-LT is much faster and more versatile than the previous CAD-score implementations. It is designed to work for both global comparisons and for localized analyses such as interfaces and binding sites.</p>
<h2 id="cad-score-definition-for-residue-residue-contacts">CAD-score definition for residue-residue contacts</h2>
<p>Given reference structure <span class="math inline"><em>T</em></span> (target) and structure to be compared, <span class="math inline"><em>M</em></span> (model), let <span class="math inline"><em>G</em></span> denote the set of all the pairs of residues <span class="math inline">(<em>i</em>,<em>j</em>)</span> that have a non-zero contact area <span class="math inline"><em>T</em><sub>(<em>i</em>,<em>j</em>)</sub></span> in the target structure. Then for every residue pair <span class="math inline">(<em>i</em>,<em>j</em>) ∈ <em>G</em></span> the corresponding contact area <span class="math inline"><em>M</em><sub>(<em>i</em>,<em>j</em>)</sub></span> in the model is calculated. <span class="math inline"><em>M</em><sub>(<em>i</em>,<em>j</em>)</sub></span> is assigned zero if there is no contact between residues <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span> in the model or if either residue (<span class="math inline"><em>i</em></span> or <span class="math inline"><em>j</em></span>) is missing from the model. The CAD-score for the model structure is then defined as:</p>
<p><span class="math display">$$\text{CAD-score}(G)=1-\frac {\sum_{(i,j) \in G} \min(|T_{(i,j)}-M_{(i,j)}|,T_{(i,j)})} {\sum_{(i,j) \in G}T_{(i,j)} }$$</span></p>
<p>CAD-score values are always within the <span class="math inline">[0,1]</span> range. If model and target structures are identical, <span class="math inline">CAD-score(<em>G</em>) = 1</span>. At the other extreme, if not a single contact is reproduced with sufficient accuracy, <span class="math inline">CAD-score(<em>G</em>) = 0</span>.</p>
<p>More local scores (e.g. interface-scores or per-residue scores) are calculated by applying the CAD-score formula to a restricted subset of <span class="math inline"><em>G</em></span>.</p>
<h2 id="cad-score-definition-for-other-types-of-areas">CAD-score definition for other types of areas</h2>
<p>If areas in a target and in a model have identifiers, and those identifiers can be matched, then CAD-score can be calculated. For example, if we want to compare solvent-accessible (SAS) areas, we can assign identifiers to such areas. Let us formulate the CAD-score definition for per-residue SAS areas in a manner similar to the CAD-score definition for residue-residue contacts.</p>
<p>Given reference structure <span class="math inline"><em>T</em></span> (target) and structure to be compared, <span class="math inline"><em>M</em></span> (model), let <span class="math inline"><em>G</em></span> denote the set of all the residues <span class="math inline"><em>i</em></span> that have a non-zero SAS area <span class="math inline"><em>T</em><sub>(<em>i</em>)</sub></span> in the target structure. Then for every residue <span class="math inline"><em>i</em> ∈ <em>G</em></span> the corresponding SAS area <span class="math inline"><em>M</em><sub>(<em>i</em>)</sub></span> in the model is calculated. <span class="math inline"><em>M</em><sub>(<em>i</em>)</sub></span> is assigned zero if there is no SAS for residue <span class="math inline"><em>i</em></span> in the model or if residue <span class="math inline"><em>i</em></span> is missing from the model. The SAS-based CAD-score for the model structure is then defined as:</p>
<p><span class="math display">$$\text{CAD-score}(G)=1-\frac {\sum_{i \in G} \min(|T_{(i)}-M_{(i)}|,T_{(i)})} {\sum_{i \in G}T_{(i)} }$$</span></p>
<p>Similarly we can calculate CAD-score for per-residue binding site areas produced by accumulating contact areas.</p>
<p>Overall, on CAD-score-LT, three types of areas can be analyzed:</p>
<ul>
<li><em>“contact”</em> - Voronoi tessellation-derived iteratomic contact areas.</li>
<li><em>“SAS”</em> - solvent-accessible surface areas.</li>
<li><em>“site”</em> - Voronoi tessellation-derived binding site areas.</li>
</ul>
<p>Areas of every type can be assessed on three levels of detail:</p>
<ul>
<li><em>“atom”</em> - atom-atom contact areas, atom-level site areas, atom-level SAS areas</li>
<li><em>“residue”</em> - residue-residue contact areas, residue-level site areas, residue-level SAS areas</li>
<li><em>“chain”</em> - chain-chain contact areas, residue-level site areas, residue-level SAS areas</li>
</ul>
<p>All the areas are intitially computed on the atom level. Then, if needed, the atom-level areas are aggregated on residue or/and chain levels.</p>
<h1 id="quickest-install-guide">Quickest install guide</h1>
<p>Since CAD-score-LT version 0.8.106, universal binary execuitables of CAD-score-LT built with the <a href="https://github.com/jart/cosmopolitan">Cosmopolitan Libc toolkit</a> are provided.</p>
<p>To download and prepare the latest released cosmopolitan executable, run the following commands in a shell environment (e.g. a Bash shell):</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="st">&#39;https://github.com/kliment-olechnovic/voronota/releases/download/v1.29.4602/cosmopolitan_cadscore-lt_v0.8.105.exe&#39;</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> cosmopolitan_cadscore-lt_v0.8.105.exe cadscore-lt</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">chmod</span> +x cadscore-lt</span></code></pre></div>
<p>In case of a PowerShell environment in Windows 8, the setup can be done with a single command:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Invoke-WebRequest</span> <span class="at">-Uri</span> <span class="st">&#39;https://github.com/kliment-olechnovic/voronota/releases/download/v1.29.4602/cosmopolitan_cadscore-lt_v0.8.105.exe&#39;</span> <span class="at">-OutFile</span> cadscore-lt.exe</span></code></pre></div>
<h1 id="getting-the-latest-version">Getting the latest version</h1>
<p>Download the latest CAD-score-LT source archive from the official downloads page: <a href="https://github.com/kliment-olechnovic/voronota/releases">https://github.com/kliment-olechnovic/voronota/releases</a>.</p>
<p>The cadscore-lt executable can be built from the provided source code to work on any modern Linux, macOS or Windows operating systems.</p>
<h1 id="building-the-command-line-tool-from-source-code">Building the command-line tool from source code</h1>
<h2 id="requirements">Requirements</h2>
<p>CAD-score-LT has no required external dependencies, only a C++17-compliant compiler is needed to build it.</p>
<h2 id="using-cmake">Using CMake</h2>
<p>You can build using CMake for makefile generation.</p>
<p>Change to the CAD-score-LT directory with <code>CMakeLists.txt</code> file, then run the sequence of commands:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cmake</span> ./</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">make</span></span></code></pre></div>
<p>Alternatively, to keep files more organized, CMake can be run in a separate “build” directory:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> build</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> build</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cmake</span> ../</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">make</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ./cadscore-lt ../cadscore-lt</span></code></pre></div>
<h2 id="using-c-compiler-directly">Using C++ compiler directly</h2>
<p>For example, “cadscore-lt” executable can be built using GNU C++ compiler.</p>
<p>Then run the CAD-score-LT directory and run the compilation command:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">g++</span> <span class="at">-std</span><span class="op">=</span>c++14 <span class="at">-O3</span> <span class="at">-fopenmp</span> <span class="at">-I</span> ./src <span class="at">-o</span> ./cadscore-lt ./src/cadscore_lt.cpp</span></code></pre></div>
<p>Performance-boosting compiler flags can be included:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">g++</span> <span class="at">-std</span><span class="op">=</span>c++14 <span class="at">-Ofast</span> <span class="at">-march</span><span class="op">=</span>native <span class="at">-fopenmp</span> <span class="at">-I</span> ./src <span class="at">-o</span> ./cadscore-lt ./src/voronota_lt.cpp</span></code></pre></div>
<h1 id="basic-usage">Basic usage</h1>
<h2 id="listing-command-line-options">Listing command-line options</h2>
<p>Running</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-h</span></span></code></pre></div>
<p>prints an overview of the command line interface:</p>
<pre><code>CAD-score-LT version 0.8

&#39;cadscore-lt&#39; calculates CAD-score (Contact Area Difference score).

Options:
    --targets | -t                                   string     input file or directory paths for target (reference) structure files
    --models | -m                                    string     input file or directory paths for model structure files
    --processors                                     number     maximum number of OpenMP threads to use, default is 2 if OpenMP is enabled, 1 if disabled
    --recursive-directory-search                                flag to search directories recursively
    --include-heteroatoms                                       flag to include heteroatoms when reading input in PDB or mmCIF format
    --read-inputs-as-assemblies                                 flag to join multiple models into an assembly when reading a file in PDB or mmCIF format
    --radii-config-file                              string     input file path for reading atom radii assignment rules
    --probe                                          number     rolling probe radius, default is 1.4
    --restrict-raw-input                             string     selection expression to restrict input atoms before any chain renaming or residue renumbering
    --reference-sequences-file                       string     input file path for reference sequences in FASTA format
    --reference-stoichiometry                        numbers    list of stoichiometry values to apply when mapping chains to reference sequences
    --restrict-processed-input                       string     selection expression to restrict input atoms after all chain renaming and residue renumbering
    --subselect-contacts                             string     selection expression to restrict contact area descriptors to score, default is &#39;[-min-sep 1]&#39;
    --subselect-atoms                                string     selection expression to restrict atom SAS and site area descriptors to score, default is &#39;[]&#39;
    --conflate-atom-types                                       flag to conflate known equivalent atom types
    --conflation-config-file                         string     input file path for reading atom name conflation rules
    --scoring-types                                  strings    scoring types (&#39;contacts&#39;, &#39;SAS&#39;, &#39;sites&#39;), default is &#39;contacts&#39;
    --scoring-levels                                 strings    scoring levels (&#39;atom&#39;, &#39;residue&#39;, &#39;chain&#39;), default is &#39;residue&#39;
    --local-output-formats                           strings    list of formats (can include &#39;table&#39;, &#39;pdb&#39;, &#39;mmcif&#39;, &#39;contactmap&#39;, &#39;graphics-pymol&#39;, &#39;graphics-chimera&#39;)
    --local-output-levels                            strings    list of output levels (can include &#39;atom&#39;, &#39;residue&#39;, &#39;chain&#39;), default is &#39;residue&#39;
    --consider-residue-names                                    flag to include residue names in residue and atom identifiers, making mapping more strict
    --binarize-areas                                            flag to binarize (convert to 0 or 1) all area values before scoring
    --remap-chains                                              flag to automatically rename chains in models to maximize residue-residue contacts global score 
    --print-paths-in-output                                     flag to print file paths instead of file base names in output
    --save-processed-inputs-mmcif                               flag to save processed input structures in mmCIF format to the output directory
    --save-processed-inputs-pdb                                 flag to save processed input structures in PDB format to the output directory
    --save-sequence-alignments                                  flag to save best alignments with reference sequences into a file in the output directory
    --quit-before-scoring                                       flag to exit before scoring but after all the input processing and saving
    --output-with-f1                                            flag to output area-based F1 scores along with CAD-scores
    --output-all-details                                        flag to output all details in tables of global and local scores
    --compact-output                                            flag to reduce size of output global scores table without removing rows
    --output-global-scores                           string     path to output table of global scores, default is &#39;_stdout&#39; to print to standard output 
    --output-dir                                     string     path to output directory for all result files
    --help | -h                                                 flag to print help info to stderr and exit

Standard output stream:
    Global scores

Standard error output stream:
    Error messages

Usage examples:

    cadscore-lt -t ./target.pdb -m ./model1.pdb ./model2.pdb

    cadscore-lt -t ./target.pdb -m ./model1.pdb ./model2.pdb --subselect-contacts &#39;[-inter-chain]&#39;</code></pre>
<h2 id="choosing-scoring-types-and-scoring-levels">Choosing scoring types and scoring levels</h2>
<p>CAD-score-LT supports several scoring types and scoring levels.</p>
<p>Scoring types (<code>--scoring-types</code>):</p>
<ul>
<li><code>contacts</code> (default): contact-area-based CAD-score (the classic CAD-score concept)</li>
<li><code>sas</code>: solvent-accessible surface area descriptors</li>
<li><code>sites</code>: binding site area descriptors</li>
</ul>
<p>Scoring levels (<code>--scoring-levels</code>):</p>
<ul>
<li><code>atom</code></li>
<li><code>residue</code> (default)</li>
<li><code>chain</code></li>
</ul>
<p>Residue-level contacts is the default mode. That is, running</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-t</span> ./target.pdb <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-m</span> ./model1.pdb</span></code></pre></div>
<p>is equivalent to running:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-t</span> ./target.pdb <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-m</span> ./model1.pdb <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--scoring-types</span> contacts <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--scoring-levels</span> residue</span></code></pre></div>
<p>Multiple options (even all of them) can be used together, for example:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">-t</span> ./target.pdb <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">-m</span> ./model1.pdb <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--scoring-types</span> contacts sas sites <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--scoring-levels</span> atom residue chain</span></code></pre></div>
<h2 id="providing-inputs">Providing inputs</h2>
<p>For input, CAD-score-LT accepts files in PDB or mmCIF formats.</p>
<p>Basic usage (one target, one model):</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb</span></code></pre></div>
<p>One target, multiple models:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb</span></code></pre></div>
<p>Read models from stdin (convenient for <code>find</code>):</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ./input/ <span class="at">-name</span> <span class="st">&#39;model*.pdb&#39;</span> <span class="kw">|</span> <span class="ex">cadscore-lt</span> <span class="at">-t</span> ./input/target.pdb</span></code></pre></div>
<p>All-vs-all mode (no <code>--target</code> or <code>-t</code> given) is entered when no targets are provided, then CAD-score-LT sets targets = models and compares all pairs (excluding self-pairs). For example:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ./input/ <span class="at">-name</span> <span class="st">&#39;*.pdb&#39;</span> <span class="kw">|</span> <span class="ex">cadscore-lt</span></span></code></pre></div>
<p>All-vs-all mode is convenient for clustering structures or their parts (e.g. inter-chain interfaces).</p>
<h2 id="outputting-a-global-scores-table">Outputting a global scores table</h2>
<p>By default, <code>cadscore-lt</code> prints a tab-separated table of global scores to stdout.</p>
<p>Useful flags:</p>
<ul>
<li><code>--output-with-f1</code> - adds area-based F1-like metrics alongside CAD scores.</li>
<li><code>--output-all-details</code> - produces a more verbose table (more columns / details).</li>
<li><code>--compact-output</code> - keeps the table smaller (without removing rows), writing auxiliary files into the directoru specified by <code>--output-dir</code>.</li>
</ul>
<p>Useful tip: use <code>column -t</code> command to align column value in a pretty way.</p>
<p>Examples:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="at">--output-with-f1</span> <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="at">--output-all-details</span> <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<h2 id="restricting-what-to-score">Restricting what to score</h2>
<p>CAD-score-LT lets you restrict:</p>
<ul>
<li>raw input atoms (atoms before any optional chain renaming and residue renumbering)</li>
<li>processed input atoms ( atoms after optional chain renaming and residue renumbering)</li>
<li>which contacts contribute to the contacts score</li>
<li>which atoms contribute to SAS and sites scoring</li>
</ul>
<p>Both atom-focused and contact focused expressions should be provided in the format of the Voronota-LT selection expressions.</p>
<h3 id="restricting-raw-input-atoms">Restricting raw input atoms</h3>
<p>This limits atoms before any chain renaming and residue renumbering:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--restrict-raw-input</span> <span class="st">&quot;[-chain A]&quot;</span> <span class="dt">\</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<h3 id="restrict-contacts">Restrict contacts</h3>
<p>Using <code>--subselect-contacts</code> is the most common way to focus CAD-score.</p>
<p>Default contact restriction is <code>[-min-sep 1]</code> (to discard contacts between atoms in the same residue).</p>
<p>Example of sidechain–sidechain contact scoring:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-min-sep 1 -atom1 [-protein-sidechain] -atom2 [-protein-sidechain]]&quot;</span> <span class="dt">\</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>Example of scoring contacts between chain A and chain B:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-a1 [-chain A] -a2 [-chain B]]&quot;</span> <span class="dt">\</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>Example of scoring contacts between any different chains:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-inter-chain]&quot;</span> <span class="dt">\</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>Example of scoring contacts between residue sets (example with ranges):</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-a1 [-chain B -rnum 39:51] -a2 [-chain B -rnum 39:66,75:87]]&quot;</span> <span class="dt">\</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>Selections can be combined with boolean expressions, as shown below:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;(([-a1 [-chain B -rnum 39:51] -a2 [-chain B -rnum 39:66]]) or ([-a1 [-chain B -rnum 39:51] -a2 [-chain B -rnum 75:87]]))&quot;</span> <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<h3 id="restrict-atoms-for-sassites">Restrict atoms for SAS/sites</h3>
<p>When scoring SAS or binding sites, you can restrict which atoms are considered using the <code>--subselect-atoms</code> optiom:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-inter-chain]&quot;</span> <span class="dt">\</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--scoring-types</span> <span class="st">&quot;sites&quot;</span> <span class="dt">\</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-atoms</span> <span class="st">&quot;[-chain A]&quot;</span> <span class="dt">\</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-all-details</span> <span class="dt">\</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<h2 id="local-scoring-outputs">Local scoring outputs</h2>
<p>CAD-score-LT can write local scores and visualization helpers into an output directory.</p>
<p>Options to use:</p>
<ul>
<li><code>--output-dir</code> DIR</li>
<li><code>--local-output-formats</code> …</li>
</ul>
<p>Formats (<code>--local-output-formats</code>) can include:</p>
<ul>
<li><code>table</code></li>
<li><code>pdb</code></li>
<li><code>mmcif</code></li>
<li><code>contactmap</code></li>
<li><code>graphics-pymol</code></li>
<li><code>graphics-chimera</code></li>
</ul>
<p>Example of writing local per-residue scores as as tables and as PDB B-factors:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-dir</span> ./results <span class="dt">\</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--local-output-formats</span> table pdb <span class="dt">\</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>Example of writing interface local scores:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-inter-chain]&quot;</span> <span class="dt">\</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output-dir</span> ./results_inter_chain <span class="dt">\</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--local-output-formats</span> table mmcif contactmap graphics-pymol <span class="dt">\</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<p>PDB/mmCIF outputs are typically easiest to inspect in molecular viewers by coloring residues by B-factor. CAD-score also sets occupancy values - 1 for every atom that contributed to any areas involved in scoring, 0 for all the other atoms. For example, when scoring an inter-chain interface only atoms involved in inter-chain contacts will have occupancy value of 1.</p>
<h2 id="chain-remapping-for-complexes">Chain remapping for complexes</h2>
<p>When scoring multimeric complexes, model chain IDs may not match target chain IDs (especially homomers).</p>
<p>CAD-score-LT can automatically rename chains in models to try to maximize the global residue–residue contacts score. For that, use the <code>--remap-chains</code> flag:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cadscore-lt</span> <span class="at">-t</span> target.pdb <span class="at">-m</span> model1.pdb model2.pdb <span class="dt">\</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--subselect-contacts</span> <span class="st">&quot;[-inter-chain]&quot;</span> <span class="dt">\</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--remap-chains</span> <span class="dt">\</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span></code></pre></div>
<h1 id="cad-score-lt-python-bindings">CAD-score-LT Python bindings</h1>
<p>The CAD-score-LT Python interface PyPI package is hosted at <a href="https://pypi.org/project/cadscorelt/">https://pypi.org/project/cadscorelt/</a>.</p>
<h2 id="installation">Installation</h2>
<p>Install with pip using this command:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install cadscorelt</span></code></pre></div>
<p>Additionally, it is recommended to have the <a href="https://pandas.pydata.org/">pandas</a> library for data analysis available in the Python environment. This allows the CAD-score result tables to be converted to <a href="https://pandas.pydata.org/docs/reference/frame.html">pandas data frames</a>. The <code>pandas</code> library can also be installed using <code>pip</code>:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install pandas</span></code></pre></div>
<p>CAD-score-LT also provides integration with some common libraries for reading macromolecular files - <a href="https://www.biotite-python.org/">Biotite</a>, <a href="https://gemmi.readthedocs.io/">Gemmi</a>, <a href="https://biopython.org/">Biopython</a>, if those libraries are available in the Python environment. They can be installed via <code>pip</code>:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install biotite</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install gemmi</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install biopython</span></code></pre></div>
<h2 id="usage-examples">Usage examples</h2>
<h3 id="basic-example">Basic example</h3>
<p>Below is an example script that calculates CAD-scores for inter-chain residue-residue contact areas, produces a table of global scores, converts that table to <code>pandas</code> data frame, and preints the top rows of the data frame:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode py"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cadscorelt</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="co"># init a CAD-score computation object</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>css <span class="op">=</span> cadscorelt.CADScoreComputer.init(subselect_contacts<span class="op">=</span><span class="st">&quot;[-inter-chain]&quot;</span>)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add a target structure</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>css.add_target_structure_from_file(<span class="st">&quot;./input/data/protein_homodimer1/target.pdb&quot;</span>)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="co"># add a target structure</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>css.add_model_structure_from_file(<span class="st">&quot;./input/data/protein_homodimer1/model1.pdb&quot;</span>)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>css.add_model_structure_from_file(<span class="st">&quot;./input/data/protein_homodimer1/model2.pdb&quot;</span>)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="co"># get a list of global scores and convert it to pandas data frame</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>df_global_scores_residue_residue <span class="op">=</span> css.get_all_cadscores_residue_residue_summarized_globally().to_pandas()</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a><span class="co"># print the first rows of the data frame</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_global_scores_residue_residue)</span></code></pre></div>
<p>Below is an example of the printed output:</p>
<pre><code>target_name model_name  CAD_score  F1_of_areas  target_area  model_area    TP_area    FP_area    FN_area renamed_chains
     target     model2   0.621922     0.774894  1047.807935  941.514533 784.870893 193.071031 262.937041              .
     target     model1   0.507319     0.639249  1047.807935  792.834440 648.098138 331.779276 399.709796              .</code></pre>
<h3 id="basic-example-using-different-structure-readers">Basic example using different structure readers</h3>
<p>Below is an example script that is similar to the previous example script, but it shows how to input structures from different sources:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode py"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cadscorelt</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># init a CAD-score computation object</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>csc <span class="op">=</span> cadscorelt.CADScoreComputer.init(subselect_contacts<span class="op">=</span><span class="st">&quot;[-inter-chain]&quot;</span>)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add a target structure read by Biotite</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> biotite.structure.io</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>structure_target <span class="op">=</span> biotite.structure.io.load_structure(<span class="st">&quot;./input/data/protein_homodimer1/target.pdb&quot;</span>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>csc.add_target_structure_from_biotite(structure_target, <span class="st">&quot;target&quot;</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co"># add a model structure read by Gemmi </span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> gemmi</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>structure_model1 <span class="op">=</span> gemmi.read_structure(<span class="st">&quot;./input/data/protein_homodimer1/model1.pdb&quot;</span>)</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>csc.add_model_structure_from_gemmi(structure_model1[<span class="dv">0</span>], <span class="st">&quot;model1&quot;</span>)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co"># add a model structure read by Biopython </span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> Bio.PDB</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>parser <span class="op">=</span> Bio.PDB.PDBParser(QUIET<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>structure_model2 <span class="op">=</span> parser.get_structure(<span class="st">&quot;id&quot;</span>, <span class="st">&quot;./input/data/protein_homodimer1/model2.pdb&quot;</span>).get_atoms()</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>csc.add_model_structure_from_biopython(structure_model2, <span class="st">&quot;model2&quot;</span>)</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a><span class="co"># get a list of global scores and convert it to pandas data frame</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>df_global_scores_residue_residue <span class="op">=</span> csc.get_all_cadscores_residue_residue_summarized_globally().to_pandas()</span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a><span class="co"># print the first rows of the data frame</span></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_global_scores_residue_residue)</span></code></pre></div>
<p>Below is an example of the printed output:</p>
<pre><code>target_name model_name  CAD_score  F1_of_areas  target_area  model_area    TP_area    FP_area    FN_area renamed_chains
     target     model2   0.621922     0.774894  1047.808013  941.514579 784.870926 193.071041 262.937087              .
     target     model1   0.507319     0.639249  1047.808013  792.834440 648.098141 331.779274 399.709873              .</code></pre>
<h3 id="advanced-example">Advanced example</h3>
<p>Below is an example script that that includes:</p>
<ul>
<li>residue-residue contact scoring (enabled by default);</li>
<li>atom-atom contact scoring;</li>
<li>automatic chain remapping to maximize grobal similarity;</li>
<li>recording local scores.</li>
</ul>
<div class="sourceCode" id="cb35"><pre class="sourceCode py"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> cadscorelt</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pathlib <span class="im">import</span> Path</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co"># to make comparison more strict, globally enable inclusion of residue names into atom and residue identifiers</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>cadscorelt.enable_considering_residue_names()</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co"># init a CAD-score computation object, enable atom-atom contact scoring, enable automatic chain remapping to maximize grobal similarity, enable recording local scores</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>csc <span class="op">=</span> cadscorelt.CADScoreComputer.init(subselect_contacts<span class="op">=</span><span class="st">&quot;[-inter-chain]&quot;</span>, score_atom_atom_contacts<span class="op">=</span><span class="va">True</span>, remap_chains<span class="op">=</span><span class="va">True</span>, record_local_scores<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="co"># set reference sequences and stoichiometry for automatic residue renumbering and chain namne assignment</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>csc.set_reference_sequences_from_file(<span class="st">&quot;./input/data/protein_heteromer1/sequences.fasta&quot;</span>)</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>csc.set_reference_stoichiometry([<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>])</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a><span class="co"># input structures from all the files in a directory</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>input_directory <span class="op">=</span> Path(<span class="st">&quot;./input/data/protein_heteromer1/structures&quot;</span>)</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> file_path <span class="kw">in</span> input_directory.iterdir():</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> file_path.is_file():</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>        csc.add_structure_from_file(<span class="bu">str</span>(file_path))</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of structure decriptors and print its top rows</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>df_structure_descriptors <span class="op">=</span> csc.get_all_structure_descriptors().to_pandas()</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of structure decriptors:&quot;</span>)</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_structure_descriptors)</span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of globals scores based on residue-residue contacts, print top rows</span></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>df_global_scores_residue_residue <span class="op">=</span> csc.get_all_cadscores_residue_residue_summarized_globally().to_pandas()</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of globals scores based on residue-residue contacts:&quot;</span>)</span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_global_scores_residue_residue)</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of globals scores based on atom-atom contacts, print top rows</span></span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>df_global_scores_atom_atom <span class="op">=</span> csc.get_all_cadscores_atom_atom_summarized_globally().to_pandas()</span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of globals scores based on atom-atom contacts:&quot;</span>)</span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_global_scores_atom_atom)</span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a><span class="co"># set placeholder variable for structure names</span></span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a>target_name<span class="op">=</span><span class="st">&quot;cf_woTemplates_model_3_multimer_v3_pred_47&quot;</span></span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a>model_name<span class="op">=</span><span class="st">&quot;cf_woTemplates_model_2_multimer_v3_pred_26&quot;</span></span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-48"><a href="#cb35-48" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of per-residue scores based on residue-residue contacts, print top rows</span></span>
<span id="cb35-49"><a href="#cb35-49" aria-hidden="true" tabindex="-1"></a>df_local_scores_per_residue <span class="op">=</span> csc.get_local_cadscores_residue_residue_summarized_per_residue(target_name, model_name).to_pandas()</span>
<span id="cb35-50"><a href="#cb35-50" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-51"><a href="#cb35-51" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of per-residue scores based on residue-residue contacts:&quot;</span>)</span>
<span id="cb35-52"><a href="#cb35-52" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-53"><a href="#cb35-53" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_local_scores_per_residue)</span>
<span id="cb35-54"><a href="#cb35-54" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-55"><a href="#cb35-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-56"><a href="#cb35-56" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of scores for every residue-residue contact, print top rows</span></span>
<span id="cb35-57"><a href="#cb35-57" aria-hidden="true" tabindex="-1"></a>df_local_scores_residue_residue <span class="op">=</span> csc.get_local_cadscores_residue_residue(target_name, model_name).to_pandas()</span>
<span id="cb35-58"><a href="#cb35-58" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-59"><a href="#cb35-59" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of scores for every residue-residue contact (CAD-score values of -1 idicate that the contact was not present in the target structure):&quot;</span>)</span>
<span id="cb35-60"><a href="#cb35-60" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-61"><a href="#cb35-61" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_local_scores_residue_residue)</span>
<span id="cb35-62"><a href="#cb35-62" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-63"><a href="#cb35-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-64"><a href="#cb35-64" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of per-atom scores based on atom-atom contacts, print top rows</span></span>
<span id="cb35-65"><a href="#cb35-65" aria-hidden="true" tabindex="-1"></a>df_local_scores_per_atom <span class="op">=</span> csc.get_local_cadscores_atom_atom_summarized_per_atom(target_name, model_name).to_pandas()</span>
<span id="cb35-66"><a href="#cb35-66" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-67"><a href="#cb35-67" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of per-atom scores based on atom-atom contacts (CAD-score values of -1 idicate that the atom had no relevant contacts in the target structure):&quot;</span>)</span>
<span id="cb35-68"><a href="#cb35-68" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-69"><a href="#cb35-69" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_local_scores_per_atom)</span>
<span id="cb35-70"><a href="#cb35-70" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-71"><a href="#cb35-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-72"><a href="#cb35-72" aria-hidden="true" tabindex="-1"></a><span class="co"># get the table of scores for every atom-atom contact, print top rows</span></span>
<span id="cb35-73"><a href="#cb35-73" aria-hidden="true" tabindex="-1"></a>df_local_scores_atom_atom <span class="op">=</span> csc.get_local_cadscores_atom_atom(target_name, model_name).to_pandas()</span>
<span id="cb35-74"><a href="#cb35-74" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-75"><a href="#cb35-75" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot; # Table of scores for every atom-atom contact (CAD-score values of -1 idicate that the contact was not present in the target structure):&quot;</span>)</span>
<span id="cb35-76"><a href="#cb35-76" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb35-77"><a href="#cb35-77" aria-hidden="true" tabindex="-1"></a>cadscorelt.print_head_of_pandas_data_frame(df_local_scores_atom_atom)</span>
<span id="cb35-78"><a href="#cb35-78" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>Below is an example of the printed output:</p>
<pre><code>
 # Table of structure decriptors:

                                                    name  is_target  is_model          renamed_chains reference_alignment
                   afm_basic_model_5_multimer_v1_pred_35       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
            afm_dropout_full_model_1_multimer_v2_pred_42       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
            afm_dropout_full_model_2_multimer_v1_pred_65       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
            afm_dropout_full_model_3_multimer_v3_pred_64       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
            afm_dropout_full_model_3_multimer_v3_pred_66       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
 afm_dropout_full_woTemplates_model_3_multimer_v1_pred_4       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
afm_dropout_full_woTemplates_model_3_multimer_v1_pred_45       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
afm_dropout_full_woTemplates_model_4_multimer_v3_pred_50       True      True B=A,C=D,D=B,E=E,F=C,G=F           available
              cf_woTemplates_model_2_multimer_v3_pred_26       True      True A=A,B=D,C=B,D=E,E=C,F=F           available
              cf_woTemplates_model_3_multimer_v3_pred_47       True      True A=A,B=D,C=B,D=E,E=C,F=F           available


 # Table of globals scores based on residue-residue contacts:

                                 target_name                                               model_name  CAD_score  F1_of_areas  target_area  model_area     TP_area     FP_area     FN_area          renamed_chains
afm_dropout_full_model_1_multimer_v2_pred_42 afm_dropout_full_woTemplates_model_4_multimer_v3_pred_50   0.847662     0.701359  3970.007175 3836.593118 3582.890995 2664.103761  387.116180 A=A;B=B;C=F;D=D;E=E;F=C
  cf_woTemplates_model_3_multimer_v3_pred_47               cf_woTemplates_model_2_multimer_v3_pred_26   0.704688     0.780891  7483.973156 6871.838277 5912.094425 1745.848929 1571.878730 A=D;B=E;C=F;D=A;E=B;F=C
  cf_woTemplates_model_2_multimer_v3_pred_26               cf_woTemplates_model_3_multimer_v3_pred_47   0.699040     0.781187  7657.943354 6855.599895 5914.330867 1569.642289 1743.612487 A=A;B=B;C=C;D=D;E=E;F=F
afm_dropout_full_model_3_multimer_v3_pred_66             afm_dropout_full_model_3_multimer_v3_pred_64   0.657842     0.675283  7313.203511 5519.154809 5124.561234 2739.763621 2188.642277 A=A;B=B;C=C;D=D;E=E;F=F
afm_dropout_full_model_3_multimer_v3_pred_64             afm_dropout_full_model_3_multimer_v3_pred_66   0.620764     0.675283  7864.324855 5420.088717 5124.561234 2188.642277 2739.763621 A=A;B=B;C=C;D=D;E=E;F=F
afm_dropout_full_model_3_multimer_v3_pred_66               cf_woTemplates_model_2_multimer_v3_pred_26   0.575097     0.645935  7313.203511 5696.007986 4835.195167 2822.748187 2478.008344 A=A;B=E;C=C;D=D;E=B;F=F
  cf_woTemplates_model_2_multimer_v3_pred_26             afm_dropout_full_model_3_multimer_v3_pred_66   0.573808     0.645935  7657.943354 5446.078054 4835.195167 2478.008344 2822.748187 A=A;B=E;C=C;D=D;E=B;F=F
afm_dropout_full_model_3_multimer_v3_pred_66               cf_woTemplates_model_3_multimer_v3_pred_47   0.566239     0.627281  7313.203511 5493.415167 4640.996876 2842.976280 2672.206635 A=D;B=B;C=F;D=A;E=E;F=C
  cf_woTemplates_model_3_multimer_v3_pred_47             afm_dropout_full_model_3_multimer_v3_pred_66   0.561418     0.627332  7483.973156 5252.699687 4641.373085 2671.830426 2842.600071 A=A;B=E;C=C;D=D;E=B;F=F
  cf_woTemplates_model_2_multimer_v3_pred_26             afm_dropout_full_model_3_multimer_v3_pred_64   0.550970     0.603414  7657.943354 5255.694013 4683.177891 3181.146964 2974.765463 A=A;B=E;C=C;D=D;E=B;F=F


 # Table of globals scores based on atom-atom contacts:

                                 target_name                                               model_name  CAD_score  F1_of_areas  target_area  model_area     TP_area     FP_area     FN_area          renamed_chains
afm_dropout_full_model_1_multimer_v2_pred_42 afm_dropout_full_woTemplates_model_4_multimer_v3_pred_50   0.694870     0.612790  3970.007175 3631.663716 3130.437751 3116.557004  839.569423 A=A;B=B;C=F;D=D;E=E;F=C
afm_dropout_full_model_3_multimer_v3_pred_66             afm_dropout_full_model_3_multimer_v3_pred_64   0.600300     0.630986  7313.203511 5287.148112 4788.403033 3075.921822 2524.800478 A=A;B=B;C=C;D=D;E=E;F=F
afm_dropout_full_model_3_multimer_v3_pred_64             afm_dropout_full_model_3_multimer_v3_pred_66   0.566073     0.630986  7864.324855 5222.053418 4788.403033 2524.800478 3075.921822 A=A;B=B;C=C;D=D;E=E;F=F
  cf_woTemplates_model_3_multimer_v3_pred_47               cf_woTemplates_model_2_multimer_v3_pred_26   0.559002     0.635331  7483.973156 5931.794420 4810.063036 2847.880319 2673.910120 A=D;B=E;C=F;D=A;E=B;F=C
  cf_woTemplates_model_2_multimer_v3_pred_26               cf_woTemplates_model_3_multimer_v3_pred_47   0.550378     0.635281  7657.943354 5972.073003 4809.683681 2674.289475 2848.259673 A=A;B=B;C=C;D=D;E=E;F=F
afm_dropout_full_model_3_multimer_v3_pred_66               cf_woTemplates_model_2_multimer_v3_pred_26   0.455307     0.527325  7313.203511 5000.847832 3947.330056 3710.613299 3365.873455 A=A;B=E;C=C;D=D;E=B;F=F
  cf_woTemplates_model_2_multimer_v3_pred_26             afm_dropout_full_model_3_multimer_v3_pred_66   0.446372     0.527325  7657.943354 4761.917277 3947.330056 3365.873455 3710.613299 A=A;B=E;C=C;D=D;E=B;F=F
  cf_woTemplates_model_2_multimer_v3_pred_26             afm_dropout_full_model_3_multimer_v3_pred_64   0.445539     0.506868  7657.943354 4677.771912 3933.871243 3930.453612 3724.072111 A=A;B=E;C=C;D=D;E=B;F=F
  cf_woTemplates_model_3_multimer_v3_pred_47             afm_dropout_full_model_3_multimer_v3_pred_64   0.439929     0.489985  7483.973156 4453.132335 3760.215370 4104.109485 3723.757786 A=D;B=B;C=F;D=A;E=E;F=C
afm_dropout_full_model_3_multimer_v3_pred_66               cf_woTemplates_model_3_multimer_v3_pred_47   0.439601     0.499394  7313.203511 4526.494827 3694.810365 3789.162791 3618.393146 A=D;B=B;C=F;D=A;E=E;F=C


 # Table of per-residue scores based on residue-residue contacts:

ID_chain  ID_rnum ID_icode  CAD_score  F1_of_areas  target_area  model_area   TP_area   FP_area   FN_area
       A        4        .   0.000000     0.000000     5.723360    0.000000  0.000000 14.123374  5.723360
       A        6        .   0.000000     0.504191     7.178632   14.612032  7.160650 14.065207  0.017982
       A       15        .   0.000000     0.000000     1.346514    0.000000  0.000000  2.051384  1.346514
       A       17        .   0.175261     0.298250    25.529513    4.474315  4.474315  0.000000 21.055197
       A       18        .   0.378808     0.293841    26.519188   10.045678 10.045678 31.810069 16.473510
       A       19        .   0.715554     0.783253    69.951082   79.263636 63.086289 28.050569  6.864793
       A       20        .   0.228410     0.399188    97.824322   75.742435 35.484263 44.473835 62.340058
       A       21        .   0.892938     0.923627    55.431386   54.675708 52.086259  5.268696  3.345127
       A       22        .   0.000000     0.277170     0.401577    2.496120  0.401577  2.094543  0.000000
       A       23        .   0.406606     0.631095    34.355429   20.906229 17.437673  3.468556 16.917756


 # Table of scores for every residue-residue contact (CAD-score values of -1 idicate that the contact was not present in the target structure):

ID1_chain  ID1_rnum ID1_icode ID2_chain  ID2_rnum ID2_icode  CAD_score  F1_of_areas  target_area  model_area  TP_area   FP_area   FN_area
        A         4         .         D         6         .   0.000000     0.000000     0.043450    0.000000 0.000000  0.000000  0.043450
        A         4         .         D        61         .   0.000000     0.000000     5.679910    0.000000 0.000000  0.000000  5.679910
        A         4         .         D       206         .  -1.000000     0.000000     0.000000    0.000000 0.000000 14.123374  0.000000
        A         6         .         D         4         .   0.000000     0.000000     0.017982    0.000000 0.000000  0.000000  0.017982
        A         6         .         D        65         .   0.000000     0.657765     7.160650   14.612032 7.160650  7.451382  0.000000
        A         6         .         D       208         .  -1.000000     0.000000     0.000000    0.000000 0.000000  6.613825  0.000000
        A        15         .         C        31         .  -1.000000     0.000000     0.000000    0.000000 0.000000  2.051384  0.000000
        A        15         .         D       137         .   0.000000     0.000000     1.346514    0.000000 0.000000  0.000000  1.346514
        A        17         .         C        31         .   0.086301     0.158889    16.995494    1.466721 1.466721  0.000000 15.528773
        A        17         .         C        32         .   0.557286     0.715714     5.396860    3.007595 3.007595  0.000000  2.389265


 # Table of per-atom scores based on atom-atom contacts (CAD-score values of -1 idicate that the atom had no relevant contacts in the target structure):

ID_chain  ID_rnum ID_icode ID_atom_name  CAD_score  F1_of_areas  target_area  model_area  TP_area   FP_area   FN_area
       A        4        .           CD  -1.000000     0.000000     0.000000    0.000000 0.000000  0.589435  0.000000
       A        4        .           CE  -1.000000     0.000000     0.000000    0.000000 0.000000  6.218582  0.000000
       A        4        .           CG  -1.000000     0.000000     0.000000    0.000000 0.000000  0.040717  0.000000
       A        4        .           NZ   0.000000     0.000000     5.723360    0.000000 0.000000  7.274641  5.723360
       A        6        .           CB  -1.000000     0.000000     0.000000    0.000000 0.000000  0.005776  0.000000
       A        6        .          CG1  -1.000000     0.000000     0.000000    0.000000 0.000000 10.924598  0.000000
       A        6        .          CG2   0.315134     0.527725     7.178632    8.956361 4.610766  5.684717  2.567866
       A       15        .           CB  -1.000000     0.000000     0.000000    0.000000 0.000000  1.960049  0.000000
       A       15        .            O   0.000000     0.000000     1.346514    0.000000 0.000000  0.091335  1.346514
       A       17        .           CB   0.057452     0.097769    25.529513    1.466721 1.466721  3.007595 24.062792


 # Table of scores for every atom-atom contact (CAD-score values of -1 idicate that the contact was not present in the target structure):

ID1_chain  ID1_rnum ID1_icode ID1_atom_name ID2_chain  ID2_rnum ID2_icode ID2_atom_name  CAD_score  F1_of_areas  target_area  model_area  TP_area  FP_area  FN_area
        A         4         .            CD         D       206         .           OE1       -1.0          0.0     0.000000         0.0      0.0 0.589435 0.000000
        A         4         .            CE         D       206         .           NE2       -1.0          0.0     0.000000         0.0      0.0 1.398847 0.000000
        A         4         .            CE         D       206         .           OE1       -1.0          0.0     0.000000         0.0      0.0 4.819735 0.000000
        A         4         .            CG         D       206         .           OE1       -1.0          0.0     0.000000         0.0      0.0 0.040717 0.000000
        A         4         .            NZ         D         6         .           CG2        0.0          0.0     0.043450         0.0      0.0 0.000000 0.043450
        A         4         .            NZ         D        61         .            CZ        0.0          0.0     0.413695         0.0      0.0 0.000000 0.413695
        A         4         .            NZ         D        61         .            OH        0.0          0.0     5.266215         0.0      0.0 0.000000 5.266215
        A         4         .            NZ         D       206         .            CD       -1.0          0.0     0.000000         0.0      0.0 0.008074 0.000000
        A         4         .            NZ         D       206         .           NE2       -1.0          0.0     0.000000         0.0      0.0 5.372747 0.000000
        A         4         .            NZ         D       206         .           OE1       -1.0          0.0     0.000000         0.0      0.0 1.893820 0.000000
</code></pre>
</body>
</html>
